// jira.js/dist/esm/serviceDesk/request.mjs
class Request {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCustomerRequests(parameters, callback) {
        const config = {
            url: '/rest/servicedeskapi/request',
            method: 'GET',
            params: {
                searchTerm: parameters?.searchTerm,
                requestStatus: parameters?.requestStatus,
                approvalStatus: parameters?.approvalStatus,
                organizationId: parameters?.organizationId,
                serviceDeskId: parameters?.serviceDeskId,
                requestTypeId: parameters?.requestTypeId,
                expand: parameters?.expand,
                start: parameters?.start,
                limit: parameters?.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createCustomerRequest(parameters, callback) {
        const config = {
            url: '/rest/servicedeskapi/request',
            method: 'POST',
            data: {
                serviceDeskId: parameters?.serviceDeskId,
                requestTypeId: parameters?.requestTypeId,
                requestFieldValues: parameters?.requestFieldValues,
                requestParticipants: parameters?.requestParticipants,
                raiseOnBehalfOf: parameters?.raiseOnBehalfOf,
                channel: parameters?.channel,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomerRequestByIdOrKey(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getApprovals(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/approval`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getApprovalById(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/approval/${parameters.approvalId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async answerApproval(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/approval/${parameters.approvalId}`,
            method: 'POST',
            data: {
                decision: parameters.decision,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAttachmentsForRequest(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/attachment`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createAttachment(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/attachment`,
            method: 'POST',
            data: {
                temporaryAttachmentIds: parameters.temporaryAttachmentIds,
                additionalComment: parameters.additionalComment,
                public: parameters.public,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAttachmentContent(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/attachment/${parameters.attachmentId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAttachmentThumbnail(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/attachment/${parameters.attachmentId}/thumbnail`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getRequestComments(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/comment`,
            method: 'GET',
            params: {
                public: parameters.public,
                internal: parameters.internal,
                expand: parameters.expand,
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createRequestComment(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/comment`,
            method: 'POST',
            data: {
                body: parameters.body,
                public: parameters.public,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getRequestCommentById(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/comment/${parameters.commentId}`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCommentAttachments(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/comment/${parameters.commentId}/attachment`,
            method: 'GET',
            headers: {
                'X-ExperimentalApi': 'opt-in',
            },
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSubscriptionStatus(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/notification`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async subscribe(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/notification`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async unsubscribe(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/notification`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getRequestParticipants(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/participant`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addRequestParticipants(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/participant`,
            method: 'POST',
            data: {
                usernames: parameters.usernames,
                accountIds: parameters.accountIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeRequestParticipants(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/participant`,
            method: 'DELETE',
            data: {
                usernames: parameters.usernames,
                accountIds: parameters.accountIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSlaInformation(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/sla`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSlaInformationById(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/sla/${parameters.slaMetricId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomerRequestStatus(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/status`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomerTransitions(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/transition`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async performCustomerTransition(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.issueIdOrKey}/transition`,
            method: 'POST',
            data: {
                id: parameters.id,
                additionalComment: parameters.additionalComment,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFeedback(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.requestIdOrKey}/feedback`,
            method: 'GET',
            headers: {
                'X-ExperimentalApi': 'opt-in',
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async postFeedback(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.requestIdOrKey}/feedback`,
            method: 'POST',
            headers: {
                'X-ExperimentalApi': 'opt-in',
            },
            data: {
                type: parameters.type,
                rating: parameters.rating,
                comment: parameters.comment,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFeedback(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/request/${parameters.requestIdOrKey}/feedback`,
            method: 'DELETE',
            headers: {
                'X-ExperimentalApi': 'opt-in',
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Request };
