// jira.js/dist/esm/version2/avatars.mjs
class Avatars {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllSystemAvatars(parameters, callback) {
        const type = typeof parameters === 'string' ? parameters : parameters.type;
        const config = {
            url: `/rest/api/2/avatar/${type}/system`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvatars(parameters, callback) {
        const config = {
            url: `/rest/api/2/universal_avatar/type/${parameters.type}/owner/${parameters.entityId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async storeAvatar(parameters, callback) {
        const config = {
            url: `/rest/api/2/universal_avatar/type/${parameters.type}/owner/${parameters.entityId}`,
            method: 'POST',
            headers: {
                'X-Atlassian-Token': 'no-check',
                'Content-Type': parameters.mimeType,
            },
            params: {
                x: parameters.x,
                y: parameters.y,
                size: parameters.size ?? 0,
            },
            data: parameters.avatar,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteAvatar(parameters, callback) {
        const config = {
            url: `/rest/api/2/universal_avatar/type/${parameters.type}/owner/${parameters.owningObjectId}/avatar/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvatarImageByType(parameters, callback) {
        const type = typeof parameters === 'string' ? parameters : parameters.type;
        const config = {
            url: `/rest/api/2/universal_avatar/view/type/${type}`,
            method: 'GET',
            responseType: 'arraybuffer',
            params: {
                size: typeof parameters !== 'string' ? parameters.size : undefined,
                format: typeof parameters !== 'string' ? parameters.format : undefined,
            },
        };
        const { data: avatar, headers: { 'content-type': contentTypeWithEncoding }, } = await this.client.sendRequestFullResponse(config);
        const contentType = contentTypeWithEncoding.split(';')[0].trim();
        return this.client.handleSuccessResponse({ contentType, avatar }, callback);
    }
    async getAvatarImageByID(parameters, callback) {
        const config = {
            url: `/rest/api/2/universal_avatar/view/type/${parameters.type}/avatar/${parameters.id}`,
            method: 'GET',
            responseType: 'arraybuffer',
            params: {
                size: parameters.size,
                format: parameters.format,
            },
        };
        const { data: avatar, headers: { 'content-type': contentTypeWithEncoding }, } = await this.client.sendRequestFullResponse(config);
        const contentType = contentTypeWithEncoding.split(';')[0].trim();
        return this.client.handleSuccessResponse({ contentType, avatar }, callback);
    }
    async getAvatarImageByOwner(parameters, callback) {
        const config = {
            url: `/rest/api/2/universal_avatar/view/type/${parameters.type}/owner/${parameters.entityId}`,
            method: 'GET',
            responseType: 'arraybuffer',
            params: {
                size: parameters.size,
                format: parameters.format,
            },
        };
        const { data: avatar, headers: { 'content-type': contentTypeWithEncoding }, } = await this.client.sendRequestFullResponse(config);
        const contentType = contentTypeWithEncoding.split(';')[0].trim();
        return this.client.handleSuccessResponse({ contentType, avatar }, callback);
    }
}

export { Avatars };
