// jira.js/dist/esm/version2/filterSharing.mjs
class FilterSharing {
    client;
    constructor(client) {
        this.client = client;
    }
    async getDefaultShareScope(callback) {
        const config = {
            url: '/rest/api/2/filter/defaultShareScope',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultShareScope(parameters, callback) {
        const scope = typeof parameters === 'string' ? parameters : parameters.scope;
        const config = {
            url: '/rest/api/2/filter/defaultShareScope',
            method: 'PUT',
            data: {
                scope,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSharePermissions(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}/permission`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}/permission`,
            method: 'POST',
            data: {
                type: parameters.type,
                projectId: parameters.projectId,
                groupname: parameters.groupname,
                projectRoleId: parameters.projectRoleId,
                accountId: parameters.accountId,
                rights: parameters.rights,
                groupId: parameters.groupId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}/permission/${parameters.permissionId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}/permission/${parameters.permissionId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { FilterSharing };
