// jira.js/dist/esm/version2/issueCustomFieldValuesApps.mjs
class IssueCustomFieldValuesApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async updateMultipleCustomFieldValues(parameters, callback) {
        const config = {
            url: '/rest/api/2/app/field/value',
            method: 'POST',
            params: {
                generateChangelog: parameters.generateChangelog,
            },
            data: {
                updates: parameters.updates,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomFieldValue(parameters, callback) {
        const config = {
            url: `/rest/api/2/app/field/${parameters.fieldIdOrKey}/value`,
            method: 'PUT',
            params: {
                generateChangelog: parameters.generateChangelog,
            },
            data: {
                updates: parameters.updates,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueCustomFieldValuesApps };
