// jira.js/dist/esm/version2/issueCustomFieldConfigurationApps.mjs
class IssueCustomFieldConfigurationApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCustomFieldsConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/2/app/field/context/configuration/list',
            method: 'POST',
            params: {
                id: parameters?.id,
                fieldContextId: parameters?.fieldContextId,
                issueId: parameters?.issueId,
                projectKeyOrId: parameters?.projectKeyOrId,
                issueTypeId: parameters?.issueTypeId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
            data: {
                fieldIdsOrKeys: parameters?.fieldIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/2/app/field/${parameters.fieldIdOrKey}/context/configuration`,
            method: 'GET',
            params: {
                id: parameters.id,
                fieldContextId: parameters.fieldContextId,
                issueId: parameters.issueId,
                projectKeyOrId: parameters.projectKeyOrId,
                issueTypeId: parameters.issueTypeId,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/2/app/field/${parameters.fieldIdOrKey}/context/configuration`,
            method: 'PUT',
            data: {
                configurations: parameters.configurations,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueCustomFieldConfigurationApps };
