// jira.js/dist/esm/version2/issueProperties.mjs
class IssueProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async bulkSetIssuesProperties(parameters, callback) {
        const config = {
            url: '/rest/api/2/issue/properties',
            method: 'POST',
            data: {
                entitiesIds: parameters?.entitiesIds,
                properties: parameters?.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkSetIssuePropertiesByIssue(parameters, callback) {
        const config = {
            url: '/rest/api/2/issue/properties/multi',
            method: 'POST',
            data: {
                issues: parameters?.issues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkSetIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: {
                value: parameters.value,
                expression: parameters.expression,
                filter: parameters.filter,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkDeleteIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/properties/${parameters.propertyKey}`,
            method: 'DELETE',
            data: {
                entityIds: parameters.entityIds,
                currentValue: parameters.currentValue,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuePropertyKeys(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueProperties };
