// jira.js/dist/esm/version2/issueFieldConfigurations.mjs
class IssueFieldConfigurations {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllFieldConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfiguration',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                isDefault: parameters?.isDefault,
                query: parameters?.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createFieldConfiguration(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfiguration',
            method: 'POST',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/2/fieldconfiguration/${parameters.id}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFieldConfiguration(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/fieldconfiguration/${id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationItems(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/fieldconfiguration/${id}/fields`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfigurationItems(parameters, callback) {
        const config = {
            url: `/rest/api/2/fieldconfiguration/${parameters.id}/fields`,
            method: 'PUT',
            data: {
                fieldConfigurationItems: parameters.fieldConfigurationItems,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllFieldConfigurationSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfigurationscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfigurationscheme',
            method: 'POST',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationSchemeMappings(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfigurationscheme/mapping',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                fieldConfigurationSchemeId: parameters?.fieldConfigurationSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationSchemeProjectMapping(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfigurationscheme/project',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignFieldConfigurationSchemeToProject(parameters, callback) {
        const config = {
            url: '/rest/api/2/fieldconfigurationscheme/project',
            method: 'PUT',
            data: {
                fieldConfigurationSchemeId: parameters.fieldConfigurationSchemeId,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/fieldconfigurationscheme/${parameters.id}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFieldConfigurationScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/fieldconfigurationscheme/${id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async setFieldConfigurationSchemeMapping(parameters, callback) {
        const config = {
            url: `/rest/api/2/fieldconfigurationscheme/${parameters.id}/mapping`,
            method: 'PUT',
            data: {
                mappings: parameters.mappings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeIssueTypesFromGlobalFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/fieldconfigurationscheme/${parameters.id}/mapping/delete`,
            method: 'POST',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueFieldConfigurations };
