// jira.js/dist/esm/version2/issueWatchers.mjs
class IssueWatchers {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIsWatchingIssueBulk(parameters, callback) {
        const config = {
            url: '/rest/api/2/issue/watching',
            method: 'POST',
            data: {
                issueIds: parameters?.issueIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueWatchers(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/watchers`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addWatcher(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/watchers`,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            data: parameters.accountId,
        };
        return this.client.sendRequest(config, callback);
    }
    async removeWatcher(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/watchers`,
            method: 'DELETE',
            params: {
                accountId: parameters.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueWatchers };
