// jira.js/dist/esm/version2/issueLinks.mjs
class IssueLinks {
    client;
    constructor(client) {
        this.client = client;
    }
    async linkIssues(parameters, callback) {
        const config = {
            url: '/rest/api/2/issueLink',
            method: 'POST',
            data: {
                type: parameters.type,
                inwardIssue: parameters.inwardIssue,
                outwardIssue: parameters.outwardIssue,
                comment: parameters.comment,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueLink(parameters, callback) {
        const linkId = typeof parameters === 'string' ? parameters : parameters.linkId;
        const config = {
            url: `/rest/api/2/issueLink/${linkId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueLink(parameters, callback) {
        const linkId = typeof parameters === 'string' ? parameters : parameters.linkId;
        const config = {
            url: `/rest/api/2/issueLink/${linkId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueLinks };
