// jira.js/dist/esm/version2/issueSecuritySchemes.mjs
import { paramSerializer } from '../paramSerializer.mjs';

class IssueSecuritySchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueSecuritySchemes(callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueSecurityScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes',
            method: 'POST',
            data: {
                description: parameters.description,
                levels: parameters.levels,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSecurityLevels(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/level',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: paramSerializer('id', parameters?.id),
                schemeId: paramSerializer('schemeId', parameters?.schemeId),
                onlyDefault: parameters?.onlyDefault,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultLevels(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/level/default',
            method: 'PUT',
            data: {
                defaultValues: parameters?.defaultValues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSecurityLevelMembers(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/level/member',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: paramSerializer('id', parameters?.id),
                schemeId: paramSerializer('schemeId', parameters?.schemeId),
                levelId: paramSerializer('levelId', parameters?.levelId),
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchProjectsUsingSecuritySchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/project',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                issueSecuritySchemeId: parameters?.issueSecuritySchemeId,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async associateSchemesToProjects(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/project',
            method: 'PUT',
            data: {
                oldToNewSecurityLevelMappings: parameters.oldToNewSecurityLevelMappings,
                projectId: parameters.projectId,
                schemeId: parameters.schemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchSecuritySchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuesecurityschemes/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: paramSerializer('id', parameters?.id),
                projectId: paramSerializer('projectId', parameters?.projectId),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueSecurityScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueSecurityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteSecurityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async addSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}/level`,
            method: 'PUT',
            data: {
                levels: parameters.levels,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeLevel(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}`,
            method: 'DELETE',
            params: {
                replaceWith: parameters.replaceWith,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addSecurityLevelMembers(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}/member`,
            method: 'PUT',
            data: {
                members: parameters.members,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeMemberFromSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}/member/${parameters.memberId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueSecuritySchemes };
