// jira.js/dist/esm/version2/issueTypeSchemes.mjs
class IssueTypeSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllIssueTypeSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetypescheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                orderBy: parameters?.orderBy,
                expand: parameters?.expand,
                queryString: parameters?.queryString,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueTypeScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetypescheme',
            method: 'POST',
            data: {
                name: parameters.name,
                description: parameters.description,
                defaultIssueTypeId: parameters.defaultIssueTypeId,
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeSchemesMapping(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetypescheme/mapping',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                issueTypeSchemeId: parameters?.issueTypeSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeSchemeForProjects(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetypescheme/project',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignIssueTypeSchemeToProject(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetypescheme/project',
            method: 'PUT',
            data: {
                issueTypeSchemeId: parameters.issueTypeSchemeId,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueTypeScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetypescheme/${parameters.issueTypeSchemeId}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
                defaultIssueTypeId: parameters.defaultIssueTypeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueTypeScheme(parameters, callback) {
        const issueTypeSchemeId = typeof parameters === 'string' ? parameters : parameters.issueTypeSchemeId;
        const config = {
            url: `/rest/api/2/issuetypescheme/${issueTypeSchemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async addIssueTypesToIssueTypeScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetypescheme/${parameters.issueTypeSchemeId}/issuetype`,
            method: 'PUT',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async reorderIssueTypesInIssueTypeScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetypescheme/${parameters.issueTypeSchemeId}/issuetype/move`,
            method: 'PUT',
            data: {
                issueTypeIds: parameters.issueTypeIds,
                after: parameters.after,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeIssueTypeFromIssueTypeScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetypescheme/${parameters.issueTypeSchemeId}/issuetype/${parameters.issueTypeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueTypeSchemes };
