// jira.js/dist/esm/version2/issueCustomFieldOptionsApps.mjs
class IssueCustomFieldOptionsApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllIssueFieldOptions(parameters, callback) {
        const fieldKey = typeof parameters === 'string' ? parameters : parameters.fieldKey;
        const config = {
            url: `/rest/api/2/field/${fieldKey}/option`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldKey}/option`,
            method: 'POST',
            data: {
                config: parameters.config,
                properties: parameters.properties,
                value: parameters.value,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSelectableIssueFieldOptions(parameters, callback) {
        const fieldKey = typeof parameters === 'string' ? parameters : parameters.fieldKey;
        const config = {
            url: `/rest/api/2/field/${fieldKey}/option/suggestions/edit`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
                projectId: typeof parameters !== 'string' && parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getVisibleIssueFieldOptions(parameters, callback) {
        const fieldKey = typeof parameters === 'string' ? parameters : parameters.fieldKey;
        const config = {
            url: `/rest/api/2/field/${fieldKey}/option/suggestions/search`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
                projectId: typeof parameters !== 'string' && parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldKey}/option/${parameters.optionId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldKey}/option/${parameters.optionId}`,
            method: 'PUT',
            data: {
                config: parameters.config,
                id: parameters.id,
                properties: parameters.properties,
                value: parameters.value,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldKey}/option/${parameters.optionId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async replaceIssueFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldKey}/option/${parameters.optionId}/issue`,
            method: 'DELETE',
            params: {
                replaceWith: parameters.replaceWith,
                jql: parameters.jql,
                overrideScreenSecurity: parameters.overrideScreenSecurity,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueCustomFieldOptionsApps };
