// jira.js/dist/esm/version2/issueComments.mjs
class IssueComments {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCommentsByIds(parameters, callback) {
        const config = {
            url: '/rest/api/2/comment/list',
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                ids: parameters.ids,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getComments(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/comment`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
                orderBy: typeof parameters !== 'string' && parameters.orderBy,
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addComment(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/comment`,
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                author: parameters.author,
                body: parameters.comment,
                created: parameters.created,
                id: parameters.id,
                jsdAuthorCanSeeRequest: parameters.jsdAuthorCanSeeRequest,
                jsdPublic: parameters.jsdPublic,
                parentId: parameters.parentId,
                properties: parameters.properties,
                renderedBody: parameters.renderedBody,
                self: parameters.self,
                updateAuthor: parameters.updateAuthor,
                updated: parameters.updated,
                visibility: parameters.visibility,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getComment(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/comment/${parameters.id}`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateComment(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/comment/${parameters.id}`,
            method: 'PUT',
            params: {
                notifyUsers: parameters.notifyUsers,
                overrideEditableFlag: parameters.overrideEditableFlag,
                expand: parameters.expand,
            },
            data: {
                body: parameters.comment,
                visibility: parameters.visibility,
                properties: parameters.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteComment(parameters, callback) {
        const config = {
            url: `/rest/api/2/issue/${parameters.issueIdOrKey}/comment/${parameters.id}`,
            method: 'DELETE',
            params: {
                parentId: parameters.parentId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueComments };
