// jira.js/dist/esm/version2/jQL.mjs
class JQL {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAutoComplete(callback) {
        const config = {
            url: '/rest/api/2/jql/autocompletedata',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAutoCompletePost(parameters, callback) {
        const config = {
            url: '/rest/api/2/jql/autocompletedata',
            method: 'POST',
            data: {
                includeCollapsedFields: parameters?.includeCollapsedFields,
                projectIds: parameters?.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldAutoCompleteForQueryString(parameters, callback) {
        const config = {
            url: '/rest/api/2/jql/autocompletedata/suggestions',
            method: 'GET',
            params: {
                fieldName: parameters?.fieldName,
                fieldValue: parameters?.fieldValue,
                predicateName: parameters?.predicateName,
                predicateValue: parameters?.predicateValue,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async parseJqlQueries(parameters, callback) {
        const config = {
            url: '/rest/api/2/jql/parse',
            method: 'POST',
            params: {
                validation: parameters.validation,
            },
            data: {
                queries: parameters.queries,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async migrateQueries(parameters, callback) {
        const config = {
            url: '/rest/api/2/jql/pdcleaner',
            method: 'POST',
            data: {
                queryStrings: parameters?.queryStrings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async sanitiseJqlQueries(parameters, callback) {
        const config = {
            url: '/rest/api/2/jql/sanitize',
            method: 'POST',
            data: {
                queries: parameters?.queries,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { JQL };
