// jira.js/dist/esm/version2/myself.mjs
class Myself {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPreference(parameters, callback) {
        const config = {
            url: '/rest/api/2/mypreferences',
            method: 'GET',
            params: {
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setPreference(parameters, callback) {
        const config = {
            url: '/rest/api/2/mypreferences',
            method: 'PUT',
            headers: {
                'Content-Type': typeof parameters.value === 'string' ? 'text/plain' : 'application/json',
            },
            params: {
                key: parameters.key,
            },
            data: parameters.value,
        };
        return this.client.sendRequest(config, callback);
    }
    async removePreference(parameters, callback) {
        const config = {
            url: '/rest/api/2/mypreferences',
            method: 'DELETE',
            params: {
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getLocale(callback) {
        const config = {
            url: '/rest/api/2/mypreferences/locale',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getCurrentUser(parameters, callback) {
        const config = {
            url: '/rest/api/2/myself',
            method: 'GET',
            params: {
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Myself };
