// jira.js/dist/esm/version2/permissionSchemes.mjs
class PermissionSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllPermissionSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/permissionscheme',
            method: 'GET',
            params: {
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createPermissionScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/permissionscheme',
            method: 'POST',
            params: {
                expand: parameters?.expand,
            },
            data: {
                ...parameters,
                expand: undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPermissionScheme(parameters, callback) {
        const schemeId = typeof parameters === 'string' ? parameters : parameters.schemeId;
        const config = {
            url: `/rest/api/2/permissionscheme/${schemeId}`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePermissionScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}`,
            method: 'PUT',
            params: {
                expand: parameters.expand,
            },
            data: {
                ...parameters,
                schemeId: undefined,
                expand: undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deletePermissionScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getPermissionSchemeGrants(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}/permission`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createPermissionGrant(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}/permission`,
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                holder: parameters.holder,
                id: parameters.id,
                permission: parameters.permission,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPermissionSchemeGrant(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}/permission/${parameters.permissionId}`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deletePermissionSchemeEntity(parameters, callback) {
        const config = {
            url: `/rest/api/2/permissionscheme/${parameters.schemeId}/permission/${parameters.permissionId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { PermissionSchemes };
