// jira.js/dist/esm/version2/projects.mjs
class Projects {
    client;
    constructor(client) {
        this.client = client;
    }
    async createProject(parameters, callback) {
        const config = {
            url: '/rest/api/2/project',
            method: 'POST',
            data: {
                assigneeType: parameters.assigneeType,
                avatarId: parameters.avatarId,
                categoryId: parameters.categoryId,
                description: parameters.description,
                fieldConfigurationScheme: parameters.fieldConfigurationScheme,
                issueSecurityScheme: parameters.issueSecurityScheme,
                issueTypeScheme: parameters.issueTypeScheme,
                issueTypeScreenScheme: parameters.issueTypeScreenScheme,
                key: parameters.key,
                leadAccountId: parameters.leadAccountId,
                name: parameters.name,
                notificationScheme: parameters.notificationScheme,
                permissionScheme: parameters.permissionScheme,
                projectTemplateKey: parameters.projectTemplateKey,
                projectTypeKey: parameters.projectTypeKey,
                url: parameters.url,
                workflowScheme: parameters.workflowScheme,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getRecent(parameters, callback) {
        const config = {
            url: '/rest/api/2/project/recent',
            method: 'GET',
            params: {
                expand: parameters?.expand,
                properties: parameters?.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchProjects(parameters, callback) {
        const config = {
            url: '/rest/api/2/project/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                orderBy: parameters?.orderBy,
                id: parameters?.id,
                keys: parameters?.keys,
                query: parameters?.query,
                typeKey: parameters?.typeKey,
                categoryId: parameters?.categoryId,
                action: parameters?.action,
                expand: parameters?.expand,
                status: parameters?.status,
                properties: parameters?.properties,
                propertyQuery: parameters?.propertyQuery,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProject(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
                properties: typeof parameters !== 'string' ? parameters.properties : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateProject(parameters, callback) {
        const config = {
            url: `/rest/api/2/project/${parameters.projectIdOrKey}`,
            method: 'PUT',
            params: {
                expand: parameters.expand,
            },
            data: {
                assigneeType: parameters.assigneeType,
                avatarId: parameters.avatarId,
                categoryId: parameters.categoryId,
                description: parameters.description,
                issueSecurityScheme: parameters.issueSecurityScheme,
                key: parameters.key,
                leadAccountId: parameters.leadAccountId,
                name: parameters.name,
                notificationScheme: parameters.notificationScheme,
                permissionScheme: parameters.permissionScheme,
                projectTemplateKey: parameters.projectTemplateKey,
                projectTypeKey: parameters.projectTypeKey,
                releasedProjectKeys: parameters.releasedProjectKeys,
                url: parameters.url,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProject(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}`,
            method: 'DELETE',
            params: {
                enableUndo: typeof parameters !== 'string' ? parameters.enableUndo : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async archiveProject(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/archive`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProjectAsynchronously(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/delete`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async restore(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/restore`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllStatuses(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/statuses`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getHierarchy(parameters, callback) {
        const projectId = typeof parameters === 'string' ? parameters : parameters.projectId;
        const config = {
            url: `/rest/api/2/project/${projectId}/hierarchy`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getNotificationSchemeForProject(parameters, callback) {
        const projectKeyOrId = typeof parameters === 'string' ? parameters : parameters.projectKeyOrId;
        const config = {
            url: `/rest/api/2/project/${projectKeyOrId}/notificationscheme`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Projects };
