// jira.js/dist/esm/version2/projectPermissionSchemes.mjs
class ProjectPermissionSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getProjectIssueSecurityScheme(parameters, callback) {
        const projectKeyOrId = typeof parameters === 'string' ? parameters : parameters.projectKeyOrId;
        const config = {
            url: `/rest/api/2/project/${projectKeyOrId}/issuesecuritylevelscheme`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAssignedPermissionScheme(parameters, callback) {
        const projectKeyOrId = typeof parameters === 'string' ? parameters : parameters.projectKeyOrId;
        const config = {
            url: `/rest/api/2/project/${projectKeyOrId}/permissionscheme`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignPermissionScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/project/${parameters.projectKeyOrId}/permissionscheme`,
            method: 'PUT',
            params: {
                expand: parameters.expand,
            },
            data: {
                id: parameters.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSecurityLevelsForProject(parameters, callback) {
        const projectKeyOrId = typeof parameters === 'string' ? parameters : parameters.projectKeyOrId;
        const config = {
            url: `/rest/api/2/project/${projectKeyOrId}/securitylevel`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { ProjectPermissionSchemes };
