// jira.js/dist/esm/version2/screens.mjs
class Screens {
    client;
    constructor(client) {
        this.client = client;
    }
    async getScreensForField(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/field/${fieldId}/screens`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getScreens(parameters, callback) {
        const config = {
            url: '/rest/api/2/screens',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                queryString: parameters?.queryString,
                scope: parameters?.scope,
                orderBy: parameters?.orderBy,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createScreen(parameters, callback) {
        const config = {
            url: '/rest/api/2/screens',
            method: 'POST',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addFieldToDefaultScreen(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/screens/addToDefault/${fieldId}`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateScreen(parameters, callback) {
        const config = {
            url: `/rest/api/2/screens/${parameters.screenId}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteScreen(parameters, callback) {
        const screenId = typeof parameters === 'string' ? parameters : parameters.screenId;
        const config = {
            url: `/rest/api/2/screens/${screenId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvailableScreenFields(parameters, callback) {
        const screenId = typeof parameters === 'string' ? parameters : parameters.screenId;
        const config = {
            url: `/rest/api/2/screens/${screenId}/availableFields`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Screens };
