// jira.js/dist/esm/version2/timeTracking.mjs
class TimeTracking {
    client;
    constructor(client) {
        this.client = client;
    }
    async getSelectedTimeTrackingImplementation(callback) {
        const config = {
            url: '/rest/api/2/configuration/timetracking',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async selectTimeTrackingImplementation(parameters, callback) {
        const config = {
            url: '/rest/api/2/configuration/timetracking',
            method: 'PUT',
            data: {
                key: parameters?.key,
                name: parameters?.name,
                url: parameters?.url,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvailableTimeTrackingImplementations(callback) {
        const config = {
            url: '/rest/api/2/configuration/timetracking/list',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getSharedTimeTrackingConfiguration(callback) {
        const config = {
            url: '/rest/api/2/configuration/timetracking/options',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setSharedTimeTrackingConfiguration(parameters, callback) {
        const config = {
            url: '/rest/api/2/configuration/timetracking/options',
            method: 'PUT',
            data: {
                workingHoursPerDay: parameters.workingHoursPerDay,
                workingDaysPerWeek: parameters.workingDaysPerWeek,
                timeFormat: parameters.timeFormat,
                defaultUnit: parameters.defaultUnit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { TimeTracking };
