// jira.js/dist/esm/version2/users.mjs
import { paramSerializer } from '../paramSerializer.mjs';

class Users {
    client;
    constructor(client) {
        this.client = client;
    }
    async getUser(parameters, callback) {
        const config = {
            url: '/rest/api/2/user',
            method: 'GET',
            params: {
                accountId: parameters.accountId,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createUser(parameters, callback) {
        const config = {
            url: '/rest/api/2/user',
            method: 'POST',
            data: {
                emailAddress: parameters.emailAddress,
                key: parameters.key,
                name: parameters.name,
                password: parameters.password,
                products: parameters.products,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeUser(parameters, callback) {
        const config = {
            url: '/rest/api/2/user',
            method: 'DELETE',
            params: {
                accountId: parameters.accountId,
                username: parameters.username,
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkGetUsers(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/bulk',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                accountId: paramSerializer('accountId', parameters.accountId),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkGetUsersMigration(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/bulk/migration',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                username: paramSerializer('username', parameters.username),
                key: paramSerializer('key', parameters.key),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUserDefaultColumns(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/columns',
            method: 'GET',
            params: {
                accountId: parameters?.accountId,
                username: parameters?.username,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setUserColumns(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/columns',
            method: 'PUT',
            params: {
                accountId: parameters.accountId,
            },
            data: parameters.columns,
        };
        return this.client.sendRequest(config, callback);
    }
    async resetUserColumns(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/columns',
            method: 'DELETE',
            params: {
                accountId: parameters.accountId,
                username: parameters.username,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUserEmail(parameters, callback) {
        const accountId = typeof parameters === 'string' ? parameters : parameters.accountId;
        const config = {
            url: '/rest/api/2/user/email',
            method: 'GET',
            params: {
                accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUserEmailBulk(parameters, callback) {
        const accountId = typeof parameters === 'string' ? parameters : parameters.accountId;
        const config = {
            url: '/rest/api/2/user/email/bulk',
            method: 'GET',
            params: {
                accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUserGroups(parameters, callback) {
        const config = {
            url: '/rest/api/2/user/groups',
            method: 'GET',
            params: {
                accountId: parameters.accountId,
                username: parameters.username,
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllUsersDefault(parameters, callback) {
        const config = {
            url: '/rest/api/2/users',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllUsers(parameters, callback) {
        const config = {
            url: '/rest/api/2/users/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Users };
