// jira.js/dist/esm/version2/webhooks.mjs
class Webhooks {
    client;
    constructor(client) {
        this.client = client;
    }
    async getDynamicWebhooksForApp(parameters, callback) {
        const config = {
            url: '/rest/api/2/webhook',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async registerDynamicWebhooks(parameters, callback) {
        const config = {
            url: '/rest/api/2/webhook',
            method: 'POST',
            data: {
                webhooks: parameters.webhooks,
                url: parameters.url,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWebhookById(parameters, callback) {
        const config = {
            url: '/rest/api/2/webhook',
            method: 'DELETE',
            data: {
                webhookIds: parameters.webhookIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFailedWebhooks(parameters, callback) {
        const config = {
            url: '/rest/api/2/webhook/failed',
            method: 'GET',
            params: {
                maxResults: parameters?.maxResults,
                after: parameters?.after,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async refreshWebhooks(parameters, callback) {
        const config = {
            url: '/rest/api/2/webhook/refresh',
            method: 'PUT',
            data: {
                webhookIds: parameters.webhookIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Webhooks };
