// jira.js/dist/esm/version2/workflows.mjs
import { paramSerializer } from '../paramSerializer.mjs';

class Workflows {
    client;
    constructor(client) {
        this.client = client;
    }
    async createWorkflow(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflow',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
                statuses: parameters.statuses,
                transitions: parameters.transitions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowsPaginated(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflow/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                workflowName: paramSerializer('workflowName', parameters?.workflowName),
                expand: parameters?.expand,
                queryString: parameters?.queryString,
                orderBy: parameters?.orderBy,
                isActive: parameters?.isActive,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteInactiveWorkflow(parameters, callback) {
        const entityId = typeof parameters === 'string' ? parameters : parameters.entityId;
        const config = {
            url: `/rest/api/2/workflow/${entityId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowProjectIssueTypeUsages(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/${parameters.workflowId}/project/${parameters.projectId}/issueTypeUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectUsagesForWorkflow(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/${parameters.workflowId}/projectUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowSchemeUsagesForWorkflow(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/${parameters.workflowId}/workflowSchemes`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async readWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows',
            method: 'POST',
            params: {
                useTransitionLinksFormat: parameters.useTransitionLinksFormat,
                useApprovalConfiguration: parameters.useApprovalConfiguration,
            },
            data: {
                projectAndIssueTypes: parameters.projectAndIssueTypes,
                workflowIds: parameters.workflowIds,
                workflowNames: parameters.workflowNames,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async workflowCapabilities(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/capabilities',
            method: 'GET',
            params: {
                workflowId: parameters.workflowId,
                projectId: parameters.projectId,
                issueTypeId: parameters.issueTypeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/create',
            method: 'POST',
            data: {
                scope: parameters.scope,
                statuses: parameters.statuses,
                workflows: parameters.workflows,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async validateCreateWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/create/validation',
            method: 'POST',
            data: {
                payload: parameters.payload,
                validationOptions: parameters.validationOptions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                expand: parameters?.expand,
                queryString: parameters?.queryString,
                orderBy: parameters?.orderBy,
                scope: parameters?.scope,
                isActive: parameters?.isActive,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/update',
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                statuses: parameters.statuses,
                workflows: parameters.workflows,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async validateUpdateWorkflows(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/update/validation',
            method: 'POST',
            data: {
                payload: parameters.payload,
                validationOptions: parameters.validationOptions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async readWorkflowFromHistory(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflow/history',
            method: 'POST',
            data: {
                version: parameters.version,
                workflowId: parameters.workflowId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async listWorkflowHistory(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflow/history/list',
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                workflowId: parameters.workflowId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getDefaultEditor(callback) {
        const config = {
            url: '/rest/api/2/workflows/defaultEditor',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async readWorkflowPreviews(parameters, callback) {
        const config = {
            url: '/rest/api/2/workflows/preview',
            method: 'POST',
            data: {
                issueTypeIds: parameters.issueTypeIds,
                projectId: parameters.projectId,
                workflowIds: parameters.workflowIds,
                workflowNames: parameters.workflowNames,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Workflows };
