// jira.js/dist/esm/version2/workflowTransitionProperties.mjs
class WorkflowTransitionProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getWorkflowTransitionProperties(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/transitions/${parameters.transitionId}/properties`,
            method: 'GET',
            params: {
                includeReservedKeys: parameters.includeReservedKeys,
                key: parameters.key,
                workflowName: parameters.workflowName,
                workflowMode: parameters.workflowMode,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createWorkflowTransitionProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/transitions/${parameters.transitionId}/properties`,
            method: 'POST',
            params: {
                key: parameters.key,
                workflowName: parameters.workflowName,
                workflowMode: parameters.workflowMode,
            },
            data: {
                ...parameters,
                transitionId: undefined,
                key: undefined,
                workflowName: undefined,
                workflowMode: undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowTransitionProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/transitions/${parameters.transitionId}/properties`,
            method: 'PUT',
            params: {
                key: parameters.key,
                workflowName: parameters.workflowName,
                workflowMode: parameters.workflowMode,
            },
            data: {
                ...parameters,
                transitionId: undefined,
                key: undefined,
                workflowName: undefined,
                workflowMode: undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowTransitionProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflow/transitions/${parameters.transitionId}/properties`,
            method: 'DELETE',
            params: {
                key: parameters.key,
                workflowName: parameters.workflowName,
                workflowMode: parameters.workflowMode,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { WorkflowTransitionProperties };
