// jira.js/dist/cjs/version2/appMigration.cjs
'use strict';

class AppMigration {
    client;
    constructor(client) {
        this.client = client;
    }
    async updateIssueFields(parameters, callback) {
        const config = {
            url: '/rest/atlassian-connect/1/migration/field',
            method: 'PUT',
            headers: {
                'Atlassian-Account-Id': parameters.accountId,
                'Atlassian-Transfer-Id': parameters.transferId,
            },
            data: {
                updateValueList: parameters.updateValueList,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateEntityPropertiesValue(parameters, callback) {
        const config = {
            url: `/rest/atlassian-connect/1/migration/properties/${parameters.entityType}`,
            method: 'PUT',
            headers: {
                'Atlassian-Account-Id': parameters.accountId,
                'Atlassian-Transfer-Id': parameters.transferId,
                'Content-Type': 'application/json',
            },
            data: parameters.entities,
        };
        return this.client.sendRequest(config, callback);
    }
    async workflowRuleSearch(parameters, callback) {
        const config = {
            url: '/rest/atlassian-connect/1/migration/workflow/rule/search',
            method: 'POST',
            headers: {
                'Atlassian-Transfer-Id': parameters.transferId,
            },
            data: {
                expand: parameters.expand,
                ruleIds: parameters.ruleIds,
                workflowEntityId: parameters.workflowEntityId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.AppMigration = AppMigration;
