// jira.js/dist/cjs/version2/dashboards.cjs
'use strict';

var paramSerializer = require('../paramSerializer.cjs');

class Dashboards {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllDashboards(parameters, callback) {
        const config = {
            url: '/rest/api/2/dashboard',
            method: 'GET',
            params: {
                filter: parameters?.filter,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createDashboard(parameters, callback) {
        const config = {
            url: '/rest/api/2/dashboard',
            method: 'POST',
            params: {
                extendAdminPermissions: parameters.extendAdminPermissions,
            },
            data: {
                description: parameters.description,
                editPermissions: parameters.editPermissions,
                name: parameters.name,
                sharePermissions: parameters.sharePermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkEditDashboards(parameters, callback) {
        const config = {
            url: '/rest/api/2/dashboard/bulk/edit',
            method: 'PUT',
            data: {
                action: parameters.action,
                changeOwnerDetails: parameters.changeOwnerDetails,
                entityIds: parameters.entityIds,
                extendAdminPermissions: parameters.extendAdminPermissions,
                permissionDetails: parameters.permissionDetails,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllAvailableDashboardGadgets(callback) {
        const config = {
            url: '/rest/api/2/dashboard/gadgets',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDashboardsPaginated(parameters, callback) {
        const config = {
            url: '/rest/api/2/dashboard/search',
            method: 'GET',
            params: {
                dashboardName: parameters?.dashboardName,
                accountId: parameters?.accountId,
                groupname: parameters?.groupname,
                groupId: parameters?.groupId,
                projectId: parameters?.projectId,
                orderBy: parameters?.orderBy,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                status: parameters?.status,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllGadgets(parameters, callback) {
        const dashboardId = typeof parameters === 'string' ? parameters : parameters.dashboardId;
        const config = {
            url: `/rest/api/2/dashboard/${dashboardId}/gadget`,
            method: 'GET',
            params: {
                moduleKey: typeof parameters !== 'string' && paramSerializer.paramSerializer('moduleKey', parameters.moduleKey),
                uri: typeof parameters !== 'string' && parameters.uri,
                gadgetId: typeof parameters !== 'string' && paramSerializer.paramSerializer('gadgetId', parameters.gadgetId),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addGadget(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/gadget`,
            method: 'POST',
            data: {
                color: parameters.color,
                ignoreUriAndModuleKeyValidation: parameters.ignoreUriAndModuleKeyValidation,
                moduleKey: parameters.moduleKey,
                position: parameters.position,
                title: parameters.title,
                uri: parameters.uri,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateGadget(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/gadget/${parameters.gadgetId}`,
            method: 'PUT',
            data: {
                color: parameters.color,
                position: parameters.position,
                title: parameters.title,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeGadget(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/gadget/${parameters.gadgetId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDashboardItemPropertyKeys(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/items/${parameters.itemId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDashboardItemProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/items/${parameters.itemId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setDashboardItemProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/items/${parameters.itemId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json',
            },
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDashboardItemProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.dashboardId}/items/${parameters.itemId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDashboard(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/dashboard/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDashboard(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.id}`,
            method: 'PUT',
            params: {
                extendAdminPermissions: parameters.extendAdminPermissions,
            },
            data: {
                description: parameters.description,
                editPermissions: parameters.editPermissions,
                name: parameters.name,
                sharePermissions: parameters.sharePermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDashboard(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/dashboard/${id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async copyDashboard(parameters, callback) {
        const config = {
            url: `/rest/api/2/dashboard/${parameters.id}/copy`,
            method: 'POST',
            params: {
                extendAdminPermissions: parameters.extendAdminPermissions,
            },
            data: {
                description: parameters.description,
                editPermissions: parameters.editPermissions,
                name: parameters.name,
                sharePermissions: parameters.sharePermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Dashboards = Dashboards;
