// jira.js/dist/cjs/version2/dynamicModules.cjs
'use strict';

class DynamicModules {
    client;
    constructor(client) {
        this.client = client;
    }
    async getModules(callback) {
        const config = {
            url: '/rest/atlassian-connect/1/app/module/dynamic',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async registerModules(parameters, callback) {
        const config = {
            url: '/rest/atlassian-connect/1/app/module/dynamic',
            method: 'POST',
            data: {
                modules: parameters?.modules,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeModules(parameters, callback) {
        const config = {
            url: '/rest/atlassian-connect/1/app/module/dynamic',
            method: 'DELETE',
            params: {
                moduleKey: parameters?.moduleKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.DynamicModules = DynamicModules;
