// jira.js/dist/cjs/version2/fieldSchemes.cjs
'use strict';

class FieldSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getFieldAssociationSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes',
            method: 'GET',
            params: {
                projectId: parameters?.projectId,
                query: parameters?.query,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createFieldAssociationScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldsAssociatedWithSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/fields',
            method: 'PUT',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async removeFieldsAssociatedWithSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/fields',
            method: 'DELETE',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldAssociationSchemeItemParameters(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/fields/parameters',
            method: 'PUT',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async removeFieldAssociationSchemeItemParameters(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/fields/parameters',
            method: 'DELETE',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectsWithFieldSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/projects',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async associateProjectsToFieldAssociationSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/config/fieldschemes/projects',
            method: 'PUT',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldAssociationSchemeById(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldAssociationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFieldAssociationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async searchFieldAssociationSchemeFields(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}/fields`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                fieldId: parameters.fieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldAssociationSchemeItemParameters(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}/fields/${parameters.fieldId}/parameters`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async searchFieldAssociationSchemeProjects(parameters, callback) {
        const config = {
            url: `/rest/api/2/config/fieldschemes/${parameters.id}/projects`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.FieldSchemes = FieldSchemes;
