// jira.js/dist/cjs/version2/filters.cjs
'use strict';

class Filters {
    client;
    constructor(client) {
        this.client = client;
    }
    async createFilter(parameters, callback) {
        const config = {
            url: '/rest/api/2/filter',
            method: 'POST',
            params: {
                expand: parameters.expand,
                overrideSharePermissions: parameters.overrideSharePermissions,
            },
            data: {
                approximateLastUsed: parameters.approximateLastUsed,
                description: parameters.description,
                editPermissions: parameters.editPermissions,
                favourite: parameters.favourite,
                favouritedCount: parameters.favouritedCount,
                id: parameters.id,
                jql: parameters.jql,
                name: parameters.name,
                owner: parameters.owner,
                searchUrl: parameters.searchUrl,
                self: parameters.self,
                sharePermissions: parameters.sharePermissions,
                sharedUsers: parameters.sharedUsers,
                subscriptions: parameters.subscriptions,
                viewUrl: parameters.viewUrl,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFavouriteFilters(parameters, callback) {
        const config = {
            url: '/rest/api/2/filter/favourite',
            method: 'GET',
            params: {
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getMyFilters(parameters, callback) {
        const config = {
            url: '/rest/api/2/filter/my',
            method: 'GET',
            params: {
                expand: parameters?.expand,
                includeFavourites: parameters?.includeFavourites,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFiltersPaginated(parameters, callback) {
        const config = {
            url: '/rest/api/2/filter/search',
            method: 'GET',
            params: {
                filterName: parameters?.filterName,
                accountId: parameters?.accountId,
                groupname: parameters?.groupname,
                groupId: parameters?.groupId,
                projectId: parameters?.projectId,
                id: parameters?.id,
                orderBy: parameters?.orderBy,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                expand: parameters?.expand,
                overrideSharePermissions: parameters?.overrideSharePermissions,
                isSubstringMatch: parameters?.isSubstringMatch,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFilter(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' && parameters.expand,
                overrideSharePermissions: typeof parameters !== 'string' && parameters.overrideSharePermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFilter(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}`,
            method: 'PUT',
            params: {
                expand: parameters.expand,
                overrideSharePermissions: parameters.overrideSharePermissions,
            },
            data: {
                name: parameters.name,
                description: parameters.description,
                jql: parameters.jql,
                favourite: parameters.favourite,
                sharePermissions: parameters.sharePermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFilter(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getColumns(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}/columns`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setColumns(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}/columns`,
            method: 'PUT',
            data: parameters.columns,
        };
        return this.client.sendRequest(config, callback);
    }
    async resetColumns(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}/columns`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async setFavouriteForFilter(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}/favourite`,
            method: 'PUT',
            params: {
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFavouriteForFilter(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/filter/${id}/favourite`,
            method: 'DELETE',
            params: {
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async changeFilterOwner(parameters, callback) {
        const config = {
            url: `/rest/api/2/filter/${parameters.id}/owner`,
            method: 'PUT',
            data: {
                accountId: parameters.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Filters = Filters;
