// jira.js/dist/cjs/version2/issueCustomFieldContexts.cjs
'use strict';

class IssueCustomFieldContexts {
    client;
    constructor(client) {
        this.client = client;
    }
    async getContextsForField(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/field/${fieldId}/context`,
            method: 'GET',
            params: {
                isAnyIssueType: typeof parameters !== 'string' && parameters.isAnyIssueType,
                isGlobalContext: typeof parameters !== 'string' && parameters.isGlobalContext,
                contextId: typeof parameters !== 'string' && parameters.contextId,
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createCustomFieldContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context`,
            method: 'POST',
            data: {
                description: parameters.description,
                id: parameters.id,
                issueTypeIds: parameters.issueTypeIds,
                name: parameters.name,
                projectIds: parameters.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getDefaultValues(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/field/${fieldId}/context/defaultValue`,
            method: 'GET',
            params: {
                contextId: typeof parameters !== 'string' && parameters.contextId,
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultValues(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/defaultValue`,
            method: 'PUT',
            data: {
                defaultValues: parameters.defaultValues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeMappingsForContexts(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/field/${fieldId}/context/issuetypemapping`,
            method: 'GET',
            params: {
                contextId: typeof parameters !== 'string' && parameters.contextId,
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomFieldContextsForProjectsAndIssueTypes(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/mapping`,
            method: 'POST',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
            data: {
                mappings: parameters.mappings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectContextMapping(parameters, callback) {
        const fieldId = typeof parameters === 'string' ? parameters : parameters.fieldId;
        const config = {
            url: `/rest/api/2/field/${fieldId}/context/projectmapping`,
            method: 'GET',
            params: {
                contextId: typeof parameters !== 'string' && parameters.contextId,
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomFieldContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteCustomFieldContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async addIssueTypesToContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}/issuetype`,
            method: 'PUT',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeIssueTypesFromContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}/issuetype/remove`,
            method: 'POST',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignProjectsToCustomFieldContext(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}/project`,
            method: 'PUT',
            data: {
                projectIds: parameters.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeCustomFieldContextFromProjects(parameters, callback) {
        const config = {
            url: `/rest/api/2/field/${parameters.fieldId}/context/${parameters.contextId}/project/remove`,
            method: 'POST',
            data: {
                projectIds: parameters.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueCustomFieldContexts = IssueCustomFieldContexts;
