// jira.js/dist/cjs/version2/issueLinkTypes.cjs
'use strict';

class IssueLinkTypes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueLinkTypes(callback) {
        const config = {
            url: '/rest/api/2/issueLinkType',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueLinkType(parameters, callback) {
        const config = {
            url: '/rest/api/2/issueLinkType',
            method: 'POST',
            data: {
                id: parameters.id,
                inward: parameters.inward,
                name: parameters.name,
                outward: parameters.outward,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueLinkType(parameters, callback) {
        const issueLinkTypeId = typeof parameters === 'string' ? parameters : parameters.issueLinkTypeId;
        const config = {
            url: `/rest/api/2/issueLinkType/${issueLinkTypeId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueLinkType(parameters, callback) {
        const config = {
            url: `/rest/api/2/issueLinkType/${parameters.issueLinkTypeId}`,
            method: 'PUT',
            data: {
                id: parameters.id,
                inward: parameters.inward,
                name: parameters.name,
                outward: parameters.outward,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueLinkType(parameters, callback) {
        const issueLinkTypeId = typeof parameters === 'string' ? parameters : parameters.issueLinkTypeId;
        const config = {
            url: `/rest/api/2/issueLinkType/${issueLinkTypeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueLinkTypes = IssueLinkTypes;
