// jira.js/dist/cjs/version2/issueSecurityLevel.cjs
'use strict';

class IssueSecurityLevel {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueSecurityLevelMembers(parameters, callback) {
        const issueSecuritySchemeId = typeof parameters === 'string' ? parameters : parameters.issueSecuritySchemeId;
        const config = {
            url: `/rest/api/2/issuesecurityschemes/${issueSecuritySchemeId}/members`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
                issueSecurityLevelId: typeof parameters !== 'string' && parameters.issueSecurityLevelId,
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueSecurityLevel(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/securitylevel/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueSecurityLevel = IssueSecurityLevel;
