// jira.js/dist/cjs/version2/issueTypeProperties.cjs
'use strict';

class IssueTypeProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueTypePropertyKeys(parameters, callback) {
        const issueTypeId = typeof parameters === 'string' ? parameters : parameters.issueTypeId;
        const config = {
            url: `/rest/api/2/issuetype/${issueTypeId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueTypeProperties = IssueTypeProperties;
