// jira.js/dist/cjs/version2/issueTypes.cjs
'use strict';

class IssueTypes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueAllTypes(callback) {
        const config = {
            url: '/rest/api/2/issuetype',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueType(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetype',
            method: 'POST',
            data: {
                description: parameters.description,
                hierarchyLevel: parameters.hierarchyLevel,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypesForProject(parameters, callback) {
        const config = {
            url: '/rest/api/2/issuetype/project',
            method: 'GET',
            params: {
                projectId: parameters.projectId,
                level: parameters.level,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueType(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/issuetype/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetype/${parameters.id}`,
            method: 'PUT',
            data: {
                avatarId: parameters.avatarId,
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueType(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/issuetype/${id}`,
            method: 'DELETE',
            params: {
                alternativeIssueTypeId: typeof parameters !== 'string' ? parameters.alternativeIssueTypeId : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAlternativeIssueTypes(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/issuetype/${id}/alternatives`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueTypeAvatar(parameters, callback) {
        const config = {
            url: `/rest/api/2/issuetype/${parameters.id}/avatar2`,
            method: 'POST',
            headers: {
                'X-Atlassian-Token': 'no-check',
                'Content-Type': parameters.mimeType,
            },
            params: {
                x: parameters.x,
                y: parameters.y,
                size: parameters.size ?? 0,
            },
            data: parameters.avatar,
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueTypes = IssueTypes;
