// jira.js/dist/cjs/version2/issueNotificationSchemes.cjs
'use strict';

class IssueNotificationSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getNotificationSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/2/notificationscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                projectId: parameters?.projectId,
                onlyDefault: parameters?.onlyDefault,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createNotificationScheme(parameters, callback) {
        const config = {
            url: '/rest/api/2/notificationscheme',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
                notificationSchemeEvents: parameters.notificationSchemeEvents,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getNotificationSchemeToProjectMappings(parameters, callback) {
        const config = {
            url: '/rest/api/2/notificationscheme/project',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                notificationSchemeId: parameters?.notificationSchemeId,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getNotificationScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/notificationscheme/${id}`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateNotificationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/notificationscheme/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addNotifications(parameters, callback) {
        const config = {
            url: `/rest/api/2/notificationscheme/${parameters.id}/notification`,
            method: 'PUT',
            data: {
                notificationSchemeEvents: parameters.notificationSchemeEvents,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteNotificationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/notificationscheme/${parameters.notificationSchemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async removeNotificationFromNotificationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/2/notificationscheme/${parameters.notificationSchemeId}/notification/${parameters.notificationId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueNotificationSchemes = IssueNotificationSchemes;
