// jira.js/dist/cjs/version2/issueVotes.cjs
'use strict';

class IssueVotes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getVotes(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/votes`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addVote(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/votes`,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeVote(parameters, callback) {
        const issueIdOrKey = typeof parameters === 'string' ? parameters : parameters.issueIdOrKey;
        const config = {
            url: `/rest/api/2/issue/${issueIdOrKey}/votes`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueVotes = IssueVotes;
