// jira.js/dist/cjs/version2/plans.cjs
'use strict';

class Plans {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPlans(parameters, callback) {
        const config = {
            url: '/rest/api/2/plans/plan',
            method: 'GET',
            params: {
                includeTrashed: parameters?.includeTrashed,
                includeArchived: parameters?.includeArchived,
                cursor: parameters?.cursor,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createPlan(parameters, callback) {
        const config = {
            url: '/rest/api/2/plans/plan',
            method: 'POST',
            params: {
                useGroupId: parameters.useGroupId,
            },
            data: {
                crossProjectReleases: parameters.crossProjectReleases,
                customFields: parameters.customFields,
                exclusionRules: parameters.exclusionRules,
                issueSources: parameters.issueSources,
                leadAccountId: parameters.leadAccountId,
                name: parameters.name,
                permissions: parameters.permissions,
                scheduling: parameters.scheduling,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPlan(parameters, callback) {
        const config = {
            url: `/rest/api/2/plans/plan/${parameters.planId}`,
            method: 'GET',
            params: {
                useGroupId: parameters.useGroupId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePlan(parameters, callback) {
        const config = {
            url: `/rest/api/2/plans/plan/${parameters.planId}`,
            method: 'PUT',
            params: {
                useGroupId: parameters.useGroupId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async archivePlan(parameters, callback) {
        const config = {
            url: `/rest/api/2/plans/plan/${parameters.planId}/archive`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async duplicatePlan(parameters, callback) {
        const config = {
            url: `/rest/api/2/plans/plan/${parameters.planId}/duplicate`,
            method: 'POST',
            data: {
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async trashPlan(parameters, callback) {
        const config = {
            url: `/rest/api/2/plans/plan/${parameters.planId}/trash`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Plans = Plans;
