// jira.js/dist/cjs/version2/projectKeyAndNameValidation.cjs
'use strict';

class ProjectKeyAndNameValidation {
    client;
    constructor(client) {
        this.client = client;
    }
    async validateProjectKey(parameters, callback) {
        const key = typeof parameters === 'string' ? parameters : parameters?.key;
        const config = {
            url: '/rest/api/2/projectvalidate/key',
            method: 'GET',
            params: {
                key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getValidProjectKey(parameters, callback) {
        const key = typeof parameters === 'string' ? parameters : parameters?.key;
        const config = {
            url: '/rest/api/2/projectvalidate/validProjectKey',
            method: 'GET',
            params: {
                key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getValidProjectName(parameters, callback) {
        const name = typeof parameters === 'string' ? parameters : parameters.name;
        const config = {
            url: '/rest/api/2/projectvalidate/validProjectName',
            method: 'GET',
            params: {
                name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectKeyAndNameValidation = ProjectKeyAndNameValidation;
