// jira.js/dist/cjs/version2/projectVersions.cjs
'use strict';

class ProjectVersions {
    client;
    constructor(client) {
        this.client = client;
    }
    async getProjectVersionsPaginated(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/version`,
            method: 'GET',
            params: {
                startAt: typeof parameters !== 'string' ? parameters.startAt : undefined,
                maxResults: typeof parameters !== 'string' ? parameters.maxResults : undefined,
                orderBy: typeof parameters !== 'string' ? parameters.orderBy : undefined,
                query: typeof parameters !== 'string' ? parameters.query : undefined,
                status: typeof parameters !== 'string' ? parameters.status : undefined,
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectVersions(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/2/project/${projectIdOrKey}/versions`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createVersion(parameters, callback) {
        const config = {
            url: '/rest/api/2/version',
            method: 'POST',
            data: {
                approvers: parameters.approvers,
                archived: parameters.archived,
                description: parameters.description,
                driver: parameters.driver,
                expand: parameters.expand,
                id: parameters.id,
                issuesStatusForFixVersion: parameters.issuesStatusForFixVersion,
                moveUnfixedIssuesTo: parameters.moveUnfixedIssuesTo,
                name: parameters.name,
                operations: parameters.operations,
                overdue: parameters.overdue,
                projectId: parameters.projectId,
                releaseDate: parameters.releaseDate,
                released: parameters.released,
                self: parameters.self,
                startDate: parameters.startDate,
                userReleaseDate: parameters.userReleaseDate,
                userStartDate: parameters.userStartDate,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getVersion(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/version/${id}`,
            method: 'GET',
            params: {
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateVersion(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}`,
            method: 'PUT',
            data: {
                approvers: parameters.approvers,
                driver: parameters.driver,
                expand: parameters.expand,
                description: parameters.description,
                name: parameters.name,
                archived: parameters.archived,
                released: parameters.released,
                startDate: parameters.startDate,
                releaseDate: parameters.releaseDate,
                projectId: parameters.projectId,
                moveUnfixedIssuesTo: parameters.moveUnfixedIssuesTo,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async mergeVersions(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/mergeto/${parameters.moveIssuesTo}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async moveVersion(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/move`,
            method: 'POST',
            data: {
                after: parameters.after,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getVersionRelatedIssues(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/version/${id}/relatedIssueCounts`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getRelatedWork(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/relatedwork`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createRelatedWork(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/relatedwork`,
            method: 'POST',
            data: {
                category: parameters.category,
                issueId: parameters.issueId,
                relatedWorkId: parameters.relatedWorkId,
                title: parameters.title,
                url: parameters.url,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateRelatedWork(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/relatedwork`,
            method: 'PUT',
            data: {
                category: parameters.category,
                issueId: parameters.issueId,
                relatedWorkId: parameters.relatedWorkId,
                title: parameters.title,
                url: parameters.url,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteAndReplaceVersion(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.id}/removeAndSwap`,
            method: 'POST',
            data: {
                customFieldReplacementList: parameters.customFieldReplacementList,
                moveAffectedIssuesTo: parameters.moveAffectedIssuesTo,
                moveFixIssuesTo: parameters.moveFixIssuesTo,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getVersionUnresolvedIssues(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/version/${id}/unresolvedIssueCount`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRelatedWork(parameters, callback) {
        const config = {
            url: `/rest/api/2/version/${parameters.versionId}/relatedwork/${parameters.relatedWorkId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectVersions = ProjectVersions;
