// jira.js/dist/cjs/version2/screenTabs.cjs
'use strict';

class ScreenTabs {
    client;
    constructor(client) {
        this.client = client;
    }
    async getBulkScreenTabs(parameters, callback) {
        const config = {
            url: '/rest/api/2/screens/tabs',
            method: 'GET',
            params: {
                screenId: parameters?.screenId,
                tabId: parameters?.tabId,
                startAt: parameters?.startAt,
                maxResult: parameters?.maxResult,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllScreenTabs(parameters, callback) {
        const screenId = typeof parameters === 'string' ? parameters : parameters.screenId;
        const config = {
            url: `/rest/api/2/screens/${screenId}/tabs`,
            method: 'GET',
            params: {
                projectKey: typeof parameters !== 'string' && parameters.projectKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addScreenTab(parameters, callback) {
        const config = {
            url: `/rest/api/2/screens/${parameters.screenId}/tabs`,
            method: 'POST',
            data: {
                id: parameters.id,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async renameScreenTab(parameters, callback) {
        const config = {
            url: `/rest/api/2/screens/${parameters.screenId}/tabs/${parameters.tabId}`,
            method: 'PUT',
            data: {
                id: parameters.id,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteScreenTab(parameters, callback) {
        const config = {
            url: `/rest/api/2/screens/${parameters.screenId}/tabs/${parameters.tabId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async moveScreenTab(parameters, callback) {
        const config = {
            url: `/rest/api/2/screens/${parameters.screenId}/tabs/${parameters.tabId}/move/${parameters.pos}`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ScreenTabs = ScreenTabs;
