// jira.js/dist/cjs/version2/status.cjs
'use strict';

var paramSerializer = require('../paramSerializer.cjs');

class Status {
    client;
    constructor(client) {
        this.client = client;
    }
    async getStatusesById(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: '/rest/api/2/statuses',
            method: 'GET',
            params: {
                id,
                expand: typeof parameters !== 'string' && parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createStatuses(parameters, callback) {
        const config = {
            url: '/rest/api/2/statuses',
            method: 'POST',
            data: {
                statuses: parameters.statuses,
                scope: parameters.scope,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateStatuses(parameters, callback) {
        const config = {
            url: '/rest/api/2/statuses',
            method: 'PUT',
            data: {
                statuses: parameters.statuses,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteStatusesById(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: '/rest/api/2/statuses',
            method: 'DELETE',
            params: {
                id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getStatusesByName(parameters, callback) {
        const config = {
            url: '/rest/api/2/statuses/byNames',
            method: 'GET',
            params: {
                name: paramSerializer.paramSerializer('name', parameters.name),
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async search(parameters, callback) {
        const config = {
            url: '/rest/api/2/statuses/search',
            method: 'GET',
            params: {
                expand: parameters?.expand,
                projectId: parameters?.projectId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                searchString: parameters?.searchString,
                statusCategory: parameters?.statusCategory,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectIssueTypeUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/2/statuses/${parameters.statusId}/project/${parameters.projectId}/issueTypeUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/2/statuses/${parameters.statusId}/projectUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/2/statuses/${parameters.statusId}/workflowUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Status = Status;
