// jira.js/dist/cjs/version2/uIModificationsApps.cjs
'use strict';

class UIModificationsApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async getUiModifications(parameters, callback) {
        const config = {
            url: '/rest/api/2/uiModifications',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createUiModification(parameters, callback) {
        const config = {
            url: '/rest/api/2/uiModifications',
            method: 'POST',
            data: {
                name: parameters.name,
                description: parameters.description,
                data: parameters.data,
                contexts: parameters.contexts,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateUiModification(parameters, callback) {
        const config = {
            url: `/rest/api/2/uiModifications/${parameters.uiModificationId}`,
            method: 'PUT',
            data: {
                contexts: parameters.contexts,
                data: parameters.data,
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteUiModification(parameters, callback) {
        const uiModificationId = typeof parameters === 'string' ? parameters : parameters.uiModificationId;
        const config = {
            url: `/rest/api/2/uiModifications/${uiModificationId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.UIModificationsApps = UIModificationsApps;
