// jira.js/dist/cjs/version2/workflowSchemeDrafts.cjs
'use strict';

class WorkflowSchemeDrafts {
    client;
    constructor(client) {
        this.client = client;
    }
    async createWorkflowSchemeDraftFromParent(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/createdraft`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowSchemeDraft(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/draft`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowSchemeDraft(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
                defaultWorkflow: parameters.defaultWorkflow,
                issueTypeMappings: parameters.issueTypeMappings,
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowSchemeDraft(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/draft`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDraftDefaultWorkflow(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/draft/default`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDraftDefaultWorkflow(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/default`,
            method: 'PUT',
            data: {
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                workflow: parameters.workflow,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDraftDefaultWorkflow(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/draft/default`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowSchemeDraftIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/issuetype/${parameters.issueType}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setWorkflowSchemeDraftIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/issuetype/${parameters.issueType}`,
            method: 'PUT',
            data: parameters.details,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowSchemeDraftIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/issuetype/${parameters.issueType}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async publishDraftWorkflowScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/2/workflowscheme/${id}/draft/publish`,
            method: 'POST',
            params: {
                validateOnly: typeof parameters !== 'string' && parameters.validateOnly,
            },
            data: {
                statusMappings: typeof parameters !== 'string' && parameters.statusMappings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getDraftWorkflow(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/workflow`,
            method: 'GET',
            params: {
                workflowName: parameters.workflowName,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDraftWorkflowMapping(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/workflow`,
            method: 'PUT',
            params: {
                workflowName: parameters.workflowName,
            },
            data: {
                defaultMapping: parameters.defaultMapping,
                issueTypes: parameters.issueTypes,
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                workflow: parameters.workflow,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDraftWorkflowMapping(parameters, callback) {
        const config = {
            url: `/rest/api/2/workflowscheme/${parameters.id}/draft/workflow`,
            method: 'DELETE',
            params: {
                workflowName: parameters.workflowName,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.WorkflowSchemeDrafts = WorkflowSchemeDrafts;
