// jira.js/dist/cjs/agile/deployments.cjs
'use strict';

class Deployments {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitDeployments(parameters, callback) {
        const config = {
            url: '/rest/deployments/0.1/bulk',
            method: 'POST',
            data: {
                properties: parameters.properties,
                deployments: parameters.deployments,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDeploymentsByProperty(parameters, callback) {
        const config = {
            url: '/rest/deployments/0.1/bulkByProperties',
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getDeploymentByKey(parameters, callback) {
        const config = {
            url: `/rest/deployments/0.1/pipelines/${parameters.pipelineId}/environments/${parameters.environmentId}/deployments/${parameters.deploymentSequenceNumber}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDeploymentByKey(parameters, callback) {
        const config = {
            url: `/rest/deployments/0.1/pipelines/${parameters.pipelineId}/environments/${parameters.environmentId}/deployments/${parameters.deploymentSequenceNumber}`,
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getDeploymentGatingStatusByKey(parameters, callback) {
        const config = {
            url: `/rest/deployments/0.1/pipelines/${parameters.pipelineId}/environments/${parameters.environmentId}/deployments/${parameters.deploymentSequenceNumber}/gating-status`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Deployments = Deployments;
