// jira.js/dist/cjs/agile/devopsComponents.cjs
'use strict';

class DevopsComponents {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitComponents(parameters, callback) {
        const config = {
            url: '/rest/devopscomponents/1.0/bulk',
            method: 'POST',
            data: {
                properties: parameters.properties,
                components: parameters.components,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteComponentsByProperty(parameters, callback) {
        const config = {
            url: '/rest/devopscomponents/1.0/bulkByProperties',
            method: 'DELETE',
            params: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async getComponentById(parameters, callback) {
        const config = {
            url: `/rest/devopscomponents/1.0/${parameters.componentId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteComponentById(parameters, callback) {
        const config = {
            url: `/rest/devopscomponents/1.0/${parameters.componentId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.DevopsComponents = DevopsComponents;
