// jira.js/dist/cjs/agile/epic.cjs
'use strict';

class Epic {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssuesWithoutEpic(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/epic/none/issue',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                jql: parameters?.jql,
                validateQuery: parameters?.validateQuery,
                fields: parameters?.fields,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeIssuesFromEpic(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/epic/none/issue',
            method: 'POST',
            data: {
                issues: parameters?.issues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getEpic(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/epic/${parameters.epicIdOrKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async partiallyUpdateEpic(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/epic/${parameters.epicIdOrKey}`,
            method: 'POST',
            data: {
                color: parameters.color,
                done: parameters.done,
                name: parameters.name,
                summary: parameters.summary,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuesForEpic(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/epic/${parameters.epicIdOrKey}/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async moveIssuesToEpic(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/epic/${parameters.epicIdOrKey}/issue`,
            method: 'POST',
            data: {
                issues: parameters.issues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async rankEpics(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/epic/${parameters.epicIdOrKey}/rank`,
            method: 'PUT',
            data: {
                rankAfterEpic: parameters.rankAfterEpic,
                rankBeforeEpic: parameters.rankBeforeEpic,
                rankCustomFieldId: parameters.rankCustomFieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Epic = Epic;
