// jira.js/dist/cjs/agile/featureFlags.cjs
'use strict';

class FeatureFlags {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitFeatureFlags(parameters, callback) {
        const config = {
            url: '/rest/featureflags/0.1/bulk',
            method: 'POST',
            data: {
                properties: parameters.properties,
                flags: parameters.flags,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFeatureFlagsByProperty(parameters, callback) {
        const config = {
            url: '/rest/featureflags/0.1/bulkByProperties',
            method: 'DELETE',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFeatureFlagById(parameters, callback) {
        const config = {
            url: `/rest/featureflags/0.1/flag/${parameters.featureFlagId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFeatureFlagById(parameters, callback) {
        const config = {
            url: `/rest/featureflags/0.1/flag/${parameters.featureFlagId}`,
            method: 'DELETE',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.FeatureFlags = FeatureFlags;
