// jira.js/dist/cjs/agile/remoteLinks.cjs
'use strict';

class RemoteLinks {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitRemoteLinks(parameters, callback) {
        const config = {
            url: '/rest/remotelinks/1.0/bulk',
            method: 'POST',
            data: {
                properties: parameters.properties,
                remoteLinks: parameters.remoteLinks,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRemoteLinksByProperty(parameters, callback) {
        const config = {
            url: '/rest/remotelinks/1.0/bulkByProperties',
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
                params: parameters.params,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getRemoteLinkById(parameters, callback) {
        const config = {
            url: `/rest/remotelinks/1.0/remotelink/${parameters.remoteLinkId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRemoteLinkById(parameters, callback) {
        const config = {
            url: `/rest/remotelinks/1.0/remotelink/${parameters.remoteLinkId}`,
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.RemoteLinks = RemoteLinks;
