// jira.js/dist/cjs/agile/sprint.cjs
'use strict';

class Sprint {
    client;
    constructor(client) {
        this.client = client;
    }
    async createSprint(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/sprint',
            method: 'POST',
            data: {
                endDate: parameters.endDate,
                goal: parameters.goal,
                name: parameters.name,
                originBoardId: parameters.originBoardId,
                startDate: parameters.startDate,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async partiallyUpdateSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
            method: 'POST',
            data: {
                completeDate: parameters.completeDate,
                createdDate: parameters.createdDate,
                endDate: parameters.endDate,
                goal: parameters.goal,
                id: parameters.id,
                name: parameters.name,
                originBoardId: parameters.originBoardId,
                self: parameters.self,
                startDate: parameters.startDate,
                state: parameters.state,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
            method: 'PUT',
            data: {
                completeDate: parameters.completeDate,
                createdDate: parameters.createdDate,
                endDate: parameters.endDate,
                goal: parameters.goal,
                id: parameters.id,
                name: parameters.name,
                originBoardId: parameters.originBoardId,
                self: parameters.self,
                startDate: parameters.startDate,
                state: parameters.state,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuesForSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async moveIssuesToSprintAndRank(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/issue`,
            method: 'POST',
            data: {
                issues: parameters.issues,
                rankAfterIssue: parameters.rankAfterIssue,
                rankBeforeIssue: parameters.rankBeforeIssue,
                rankCustomFieldId: parameters.rankCustomFieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPropertiesKeys(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async swapSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/sprint/${parameters.sprintId}/swap`,
            method: 'POST',
            data: {
                sprintToSwapWith: parameters.sprintToSwapWith,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Sprint = Sprint;
