// jira.js/dist/esm/agile/board.mjs
class Board {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllBoards(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/board',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                type: parameters?.type,
                name: parameters?.name,
                projectKeyOrId: parameters?.projectKeyOrId,
                accountIdLocation: parameters?.accountIdLocation,
                projectLocation: parameters?.projectLocation,
                includePrivate: parameters?.includePrivate,
                negateLocationFiltering: parameters?.negateLocationFiltering,
                orderBy: parameters?.orderBy,
                expand: parameters?.expand,
                projectTypeLocation: parameters?.projectTypeLocation,
                filterId: parameters?.filterId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createBoard(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/board',
            method: 'POST',
            data: {
                filterId: parameters.filterId,
                location: parameters.location,
                name: parameters.name,
                type: parameters.type,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoardByFilterId(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/filter/${parameters.filterId}`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuesForBacklog(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/backlog`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getConfiguration(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/configuration`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getEpics(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/epic`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                done: parameters.done,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuesWithoutEpicForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/epic/none/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoardIssuesForEpic(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/epic/${parameters.epicId}/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFeaturesForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/features`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async toggleFeatures(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/features`,
            method: 'PUT',
            data: parameters.body,
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuesForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async moveIssuesToBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/issue`,
            method: 'POST',
            data: {
                issues: parameters.issues,
                rankAfterIssue: parameters.rankAfterIssue,
                rankBeforeIssue: parameters.rankBeforeIssue,
                rankCustomFieldId: parameters.rankCustomFieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjects(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/project`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectsFull(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/project/full`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoardPropertyKeys(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoardProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setBoardProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteBoardProperty(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllQuickFilters(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/quickfilter`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getQuickFilter(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/quickfilter/${parameters.quickFilterId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getReportsForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/reports`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllSprints(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/sprint`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                state: parameters.state,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBoardIssuesForSprint(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/sprint/${parameters.sprintId}/issue`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                jql: parameters.jql,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllVersions(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/board/${parameters.boardId}/version`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                released: parameters.released,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Board };
