// jira.js/dist/esm/agile/builds.mjs
class Builds {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitBuilds(parameters, callback) {
        const config = {
            url: '/rest/builds/0.1/bulk',
            method: 'POST',
            data: {
                properties: parameters.properties,
                builds: parameters.builds,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteBuildsByProperty(parameters, callback) {
        const config = {
            url: '/rest/builds/0.1/bulkByProperties',
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBuildByKey(parameters, callback) {
        const config = {
            url: `/rest/builds/0.1/pipelines/${parameters.pipelineId}/builds/${parameters.buildNumber}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteBuildByKey(parameters, callback) {
        const config = {
            url: `/rest/builds/0.1/pipelines/${parameters.pipelineId}/builds/${parameters.buildNumber}`,
            method: 'DELETE',
            params: {
                _updateSequenceNumber: parameters.updateSequenceNumber,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Builds };
