// jira.js/dist/esm/agile/developmentInformation.mjs
class DevelopmentInformation {
    client;
    constructor(client) {
        this.client = client;
    }
    async storeDevelopmentInformation(parameters, callback) {
        const config = {
            url: '/rest/devinfo/0.10/bulk',
            method: 'POST',
            data: {
                repositories: parameters.repositories,
                preventTransitions: parameters.preventTransitions,
                operationType: parameters.operationType,
                properties: parameters.properties,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getRepository(parameters, callback) {
        const config = {
            url: `/rest/devinfo/0.10/repository/${parameters.repositoryId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRepository(parameters, callback) {
        const config = {
            url: `/rest/devinfo/0.10/repository/${parameters.repositoryId}`,
            method: 'DELETE',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteByProperties(parameters, callback) {
        const config = {
            url: '/rest/devinfo/0.10/bulkByProperties',
            method: 'DELETE',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async existsByProperties(parameters, callback) {
        const config = {
            url: '/rest/devinfo/0.10/existsByProperties',
            method: 'GET',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteEntity(parameters, callback) {
        const config = {
            url: `/rest/devinfo/0.10/repository/${parameters.repositoryId}/${parameters.entityType}/${parameters.entityId}`,
            method: 'DELETE',
            params: {
                _updateSequenceId: parameters.updateSequenceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { DevelopmentInformation };
