// jira.js/dist/esm/agile/issue.mjs
class Issue {
    client;
    constructor(client) {
        this.client = client;
    }
    async rankIssues(parameters, callback) {
        const config = {
            url: '/rest/agile/1.0/issue/rank',
            method: 'PUT',
            data: {
                issues: parameters.issues,
                rankAfterIssue: parameters.rankAfterIssue,
                rankBeforeIssue: parameters.rankBeforeIssue,
                rankCustomFieldId: parameters.rankCustomFieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssue(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}`,
            method: 'GET',
            params: {
                fields: parameters.fields,
                expand: parameters.expand,
                updateHistory: parameters.updateHistory,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueEstimationForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}/estimation`,
            method: 'GET',
            params: {
                boardId: parameters.boardId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async estimateIssueForBoard(parameters, callback) {
        const config = {
            url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}/estimation`,
            method: 'PUT',
            params: {
                boardId: parameters.boardId,
            },
            data: {
                value: parameters.value,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Issue };
