// jira.js/dist/esm/agile/operations.mjs
class Operations {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitOperationsWorkspaces(parameters, callback) {
        const config = {
            url: '/rest/operations/1.0/linkedWorkspaces/bulk',
            method: 'POST',
            data: {
                workspaceIds: parameters.workspaceIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkspaces(parameters, callback) {
        const config = {
            url: '/rest/operations/1.0/linkedWorkspaces/bulk',
            method: 'DELETE',
            params: {
                workspaceIds: parameters.workspaceIds.join(','),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkspaces(parameters, callback) {
        const config = {
            url: '/rest/operations/1.0/linkedWorkspaces',
            method: 'GET',
            params: {
                workspaceId: parameters.workspaceId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitEntity(parameters, callback) {
        const config = {
            url: '/rest/operations/1.0/bulk',
            method: 'POST',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteEntityByProperty(parameters, callback) {
        const config = {
            url: '/rest/operations/1.0/bulkByProperties',
            method: 'DELETE',
            params: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async getIncidentById(parameters, callback) {
        const config = {
            url: `/rest/operations/1.0/incidents/${parameters.incidentId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIncidentById(parameters, callback) {
        const config = {
            url: `/rest/operations/1.0/incidents/${parameters.incidentId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getReviewById(parameters, callback) {
        const config = {
            url: `/rest/operations/1.0/post-incident-reviews/${parameters.reviewId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteReviewById(parameters, callback) {
        const config = {
            url: `/rest/operations/1.0/post-incident-reviews/${parameters.reviewId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Operations };
