// jira.js/dist/esm/agile/securityInformation.mjs
class SecurityInformation {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitWorkspaces(parameters, callback) {
        const config = {
            url: '/rest/security/1.0/linkedWorkspaces/bulk',
            method: 'POST',
            data: {
                workspaceIds: parameters.workspaceIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteLinkedWorkspaces(parameters, callback) {
        const config = {
            url: '/rest/security/1.0/linkedWorkspaces/bulk',
            method: 'DELETE',
            params: {
                workspaceIds: parameters.workspaceIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getLinkedWorkspaces(callback) {
        const config = {
            url: '/rest/security/1.0/linkedWorkspaces',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getLinkedWorkspaceById(parameters, callback) {
        const config = {
            url: `/rest/security/1.0/linkedWorkspaces/${parameters.workspaceId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async submitVulnerabilities(parameters, callback) {
        const config = {
            url: '/rest/security/1.0/bulk',
            method: 'POST',
            data: {
                operationType: parameters.operationType,
                properties: parameters.properties,
                vulnerabilities: parameters.vulnerabilities,
                providerMetadata: parameters.providerMetadata,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteVulnerabilitiesByProperty(parameters, callback) {
        const config = {
            url: '/rest/security/1.0/bulkByProperties',
            method: 'DELETE',
            params: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async getVulnerabilityById(parameters, callback) {
        const config = {
            url: `/rest/security/1.0/vulnerability/${parameters.vulnerabilityId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteVulnerabilityById(parameters, callback) {
        const config = {
            url: `/rest/security/1.0/vulnerability/${parameters.vulnerabilityId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { SecurityInformation };
