// jira.js/dist/cjs/config.cjs
'use strict';

var zod = require('zod');

// Authentication schemas
// const JWTSchema = z.object({
//   issuer: z.string(),
//   secret: z.string(),
//   expiryTimeSeconds: z.number().optional()
// });
const BasicAuthSchema = zod.z
    .object({
    email: zod.z.string(),
    apiToken: zod.z.string(),
})
    .strict();
const OAuth2Schema = zod.z
    .object({
    accessToken: zod.z.string(),
})
    .strict();
// Middlewares schemas
const MiddlewaresSchema = zod.z
    .object({
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    onError: zod.z.optional(zod.z.any()),
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    onResponse: zod.z.optional(zod.z.any()),
})
    .strict();
const ConfigSchema = zod.z
    .object({
    host: zod.z.string().url(),
    strictGDPR: zod.z.boolean().optional(),
    /** Adds `'X-Atlassian-Token': 'no-check'` to each request header */
    noCheckAtlassianToken: zod.z.boolean().optional(),
    baseRequestConfig: zod.z.any().optional(),
    authentication: zod.z.union([zod.z.object({ basic: BasicAuthSchema }), zod.z.object({ oauth2: OAuth2Schema })]).optional(),
    middlewares: MiddlewaresSchema.optional(),
})
    .strict();

exports.BasicAuthSchema = BasicAuthSchema;
exports.ConfigSchema = ConfigSchema;
exports.MiddlewaresSchema = MiddlewaresSchema;
exports.OAuth2Schema = OAuth2Schema;
