// jira.js/dist/cjs/createClient.cjs
'use strict';

var agileClient = require('./agile/client/agileClient.cjs');
var baseClient = require('./clients/baseClient.cjs');
require('mime');
var version2Client = require('./version2/client/version2Client.cjs');
require('zod');
require('./version3/models/projectField.cjs');
require('./version3/models/jiraStatus.cjs');
require('./version3/models/projectId.cjs');
require('./version3/models/projectIssueTypes.cjs');
require('./version3/models/statusScope.cjs');
var version3Client = require('./version3/client/version3Client.cjs');
require('./version3/parameters/getStatusesByName.cjs');
var serviceDeskClient = require('./serviceDesk/client/serviceDeskClient.cjs');

exports.ClientType = void 0;
(function (ClientType) {
    ClientType["Agile"] = "agile";
    ClientType["Version2"] = "version2";
    ClientType["Version3"] = "version3";
    ClientType["ServiceDesk"] = "serviceDesk";
})(exports.ClientType || (exports.ClientType = {}));
function createClient(clientType, config) {
    switch (clientType) {
        case exports.ClientType.Agile:
            return new agileClient.AgileClient(config);
        case exports.ClientType.Version2:
            return new version2Client.Version2Client(config);
        case exports.ClientType.Version3:
            return new version3Client.Version3Client(config);
        case exports.ClientType.ServiceDesk:
            return new serviceDeskClient.ServiceDeskClient(config);
        default:
            return new baseClient.BaseClient(config);
    }
}

exports.createClient = createClient;
