// jira.js/dist/cjs/serviceDesk/index.cjs
'use strict';

var customer = require('./customer.cjs');
var info = require('./info.cjs');
var insight = require('./insight.cjs');
var knowledgeBase = require('./knowledgeBase.cjs');
var organization = require('./organization.cjs');
var request = require('./request.cjs');
var requestType = require('./requestType.cjs');
var serviceDesk = require('./serviceDesk.cjs');
var index = require('./models/index.cjs');
var index$1 = require('./parameters/index.cjs');
var serviceDeskClient = require('./client/serviceDeskClient.cjs');



exports.Customer = customer.Customer;
exports.Info = info.Info;
exports.Insight = insight.Insight;
exports.KnowledgeBase = knowledgeBase.KnowledgeBase;
exports.Organization = organization.Organization;
exports.Request = request.Request;
exports.RequestType = requestType.RequestType;
exports.ServiceDesk = serviceDesk.ServiceDesk;
exports.ServiceDeskModels = index;
exports.ServiceDeskParameters = index$1;
exports.ServiceDeskClient = serviceDeskClient.ServiceDeskClient;
