// jira.js/dist/cjs/serviceDesk/organization.cjs
'use strict';

class Organization {
    client;
    constructor(client) {
        this.client = client;
    }
    async getOrganizations(parameters, callback) {
        const config = {
            url: '/rest/servicedeskapi/organization',
            method: 'GET',
            params: {
                start: parameters?.start,
                limit: parameters?.limit,
                accountId: parameters?.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createOrganization(parameters, callback) {
        const config = {
            url: '/rest/servicedeskapi/organization',
            method: 'POST',
            data: {
                name: parameters?.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getPropertiesKeys(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/property`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProperty(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/property/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setProperty(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/property/${parameters.propertyKey}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProperty(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/property/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getUsersInOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/user`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addUsersToOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/user`,
            method: 'POST',
            data: {
                accountIds: parameters.accountIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeUsersFromOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/organization/${parameters.organizationId}/user`,
            method: 'DELETE',
            data: {
                accountIds: parameters.accountIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getServiceDeskOrganizations(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/servicedesk/${parameters.serviceDeskId}/organization`,
            method: 'GET',
            params: {
                start: parameters.start,
                limit: parameters.limit,
                accountId: parameters.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/servicedesk/${parameters.serviceDeskId}/organization`,
            method: 'POST',
            data: {
                organizationId: parameters.organizationId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeOrganization(parameters, callback) {
        const config = {
            url: `/rest/servicedeskapi/servicedesk/${parameters.serviceDeskId}/organization`,
            method: 'DELETE',
            data: {
                organizationId: parameters.organizationId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Organization = Organization;
