// jira.js/dist/esm/version3/appProperties.mjs
class AppProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAddonProperties(parameters, callback) {
        const addonKey = typeof parameters === 'string' ? parameters : parameters.addonKey;
        const config = {
            url: `/rest/atlassian-connect/1/addons/${addonKey}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAddonProperty(parameters, callback) {
        const config = {
            url: `/rest/atlassian-connect/1/addons/${parameters.addonKey}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async putAddonProperty(parameters, callback) {
        const config = {
            url: `/rest/atlassian-connect/1/addons/${parameters.addonKey}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteAddonProperty(parameters, callback) {
        const config = {
            url: `/rest/atlassian-connect/1/addons/${parameters.addonKey}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getForgeAppPropertyKeys(callback) {
        const config = {
            url: '/rest/forge/1/app/properties',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getForgeAppProperty(parameters, callback) {
        const config = {
            url: `/rest/forge/1/app/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async putAppProperty(parameters, callback) {
        const config = {
            url: `/rest/forge/1/app/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteAppProperty(parameters, callback) {
        const config = {
            url: `/rest/forge/1/app/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { AppProperties };
