// jira.js/dist/esm/version3/groups.mjs
class Groups {
    client;
    constructor(client) {
        this.client = client;
    }
    async createGroup(parameters, callback) {
        const config = {
            url: '/rest/api/3/group',
            method: 'POST',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async removeGroup(parameters, callback) {
        const config = {
            url: '/rest/api/3/group',
            method: 'DELETE',
            params: {
                groupname: parameters.groupname,
                groupId: parameters.groupId,
                swapGroup: parameters.swapGroup,
                swapGroupId: parameters.swapGroupId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkGetGroups(parameters, callback) {
        const config = {
            url: '/rest/api/3/group/bulk',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                groupId: parameters?.groupId,
                groupName: parameters?.groupName,
                accessType: parameters?.accessType,
                applicationKey: parameters?.applicationKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUsersFromGroup(parameters, callback) {
        const config = {
            url: '/rest/api/3/group/member',
            method: 'GET',
            params: {
                groupname: parameters.groupname,
                groupId: parameters.groupId,
                includeInactiveUsers: parameters.includeInactiveUsers,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addUserToGroup(parameters, callback) {
        const config = {
            url: '/rest/api/3/group/user',
            method: 'POST',
            params: {
                groupname: parameters.groupname,
                groupId: parameters.groupId,
            },
            data: {
                accountId: parameters.accountId,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeUserFromGroup(parameters, callback) {
        const config = {
            url: '/rest/api/3/group/user',
            method: 'DELETE',
            params: {
                groupname: parameters.groupname,
                groupId: parameters.groupId,
                username: parameters.username,
                accountId: parameters.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findGroups(parameters, callback) {
        const config = {
            url: '/rest/api/3/groups/picker',
            method: 'GET',
            params: {
                query: parameters?.query,
                exclude: parameters?.exclude,
                excludeId: parameters?.excludeId,
                maxResults: parameters?.maxResults,
                caseInsensitive: parameters?.caseInsensitive,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Groups };
