// jira.js/dist/esm/version3/issueRemoteLinks.mjs
class IssueRemoteLinks {
    client;
    constructor(client) {
        this.client = client;
    }
    async getRemoteIssueLinks(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink`,
            method: 'GET',
            params: {
                globalId: parameters.globalId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createOrUpdateRemoteIssueLink(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink`,
            method: 'POST',
            data: {
                application: parameters.application,
                globalId: parameters.globalId,
                object: parameters.object,
                relationship: parameters.relationship,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRemoteIssueLinkByGlobalId(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink`,
            method: 'DELETE',
            params: {
                globalId: parameters.globalId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getRemoteIssueLinkById(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink/${parameters.linkId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateRemoteIssueLink(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink/${parameters.linkId}`,
            method: 'PUT',
            data: {
                application: parameters.application,
                globalId: parameters.globalId,
                object: parameters.object,
                relationship: parameters.relationship,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteRemoteIssueLinkById(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/remotelink/${parameters.linkId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueRemoteLinks };
