// jira.js/dist/esm/version3/issueBulkOperations.mjs
class IssueBulkOperations {
    client;
    constructor(client) {
        this.client = client;
    }
    async submitBulkDelete(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/delete',
            method: 'POST',
            data: {
                selectedIssueIdsOrKeys: parameters.selectedIssueIdsOrKeys,
                sendBulkNotification: parameters.sendBulkNotification,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBulkEditableFields(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/fields',
            method: 'GET',
            params: {
                issueIdsOrKeys: parameters.issueIdsOrKeys,
                searchText: parameters.searchText,
                endingBefore: parameters.endingBefore,
                startingAfter: parameters.startingAfter,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitBulkEdit(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/fields',
            method: 'POST',
            data: {
                editedFieldsInput: parameters.editedFieldsInput,
                selectedActions: parameters.selectedActions,
                selectedIssueIdsOrKeys: parameters.selectedIssueIdsOrKeys,
                sendBulkNotification: parameters.sendBulkNotification,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitBulkMove(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/move',
            method: 'POST',
            data: {
                sendBulkNotification: parameters.sendBulkNotification,
                targetToSourcesMapping: parameters.targetToSourcesMapping,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvailableTransitions(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/transition',
            method: 'GET',
            params: {
                issueIdsOrKeys: parameters.issueIdsOrKeys,
                endingBefore: parameters.endingBefore,
                startingAfter: parameters.startingAfter,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitBulkTransition(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/transition',
            method: 'POST',
            data: {
                bulkTransitionInputs: parameters.bulkTransitionInputs,
                sendBulkNotification: parameters.sendBulkNotification,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitBulkUnwatch(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/unwatch',
            method: 'POST',
            data: {
                selectedIssueIdsOrKeys: parameters.selectedIssueIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async submitBulkWatch(parameters, callback) {
        const config = {
            url: '/rest/api/3/bulk/issues/watch',
            method: 'POST',
            data: {
                selectedIssueIdsOrKeys: parameters.selectedIssueIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getBulkOperationProgress(parameters, callback) {
        const config = {
            url: `/rest/api/3/bulk/queue/${parameters.taskId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueBulkOperations };
