// jira.js/dist/esm/version3/issueLinkTypes.mjs
class IssueLinkTypes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueLinkTypes(callback) {
        const config = {
            url: '/rest/api/3/issueLinkType',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueLinkType(parameters, callback) {
        const config = {
            url: '/rest/api/3/issueLinkType',
            method: 'POST',
            data: {
                id: parameters.id,
                inward: parameters.inward,
                name: parameters.name,
                outward: parameters.outward,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueLinkType(parameters, callback) {
        const config = {
            url: `/rest/api/3/issueLinkType/${parameters.issueLinkTypeId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueLinkType(parameters, callback) {
        const config = {
            url: `/rest/api/3/issueLinkType/${parameters.issueLinkTypeId}`,
            method: 'PUT',
            data: {
                id: parameters.id,
                inward: parameters.inward,
                name: parameters.name,
                outward: parameters.outward,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueLinkType(parameters, callback) {
        const config = {
            url: `/rest/api/3/issueLinkType/${parameters.issueLinkTypeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueLinkTypes };
