// jira.js/dist/esm/version3/issueCommentProperties.mjs
class IssueCommentProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCommentPropertyKeys(parameters, callback) {
        const commentId = typeof parameters === 'string' ? parameters : parameters.commentId;
        const config = {
            url: `/rest/api/3/comment/${commentId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getCommentProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/comment/${parameters.commentId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setCommentProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/comment/${parameters.commentId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.property,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteCommentProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/comment/${parameters.commentId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueCommentProperties };
