// jira.js/dist/esm/version3/issueTypeProperties.mjs
class IssueTypeProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueTypePropertyKeys(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetype/${parameters.issueTypeId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueTypeProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetype/${parameters.issueTypeId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueTypeProperties };
