// jira.js/dist/esm/version3/issueSearch.mjs
class IssueSearch {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssuePickerResource(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/picker',
            method: 'GET',
            params: {
                query: parameters?.query,
                currentJQL: parameters?.currentJQL,
                currentIssueKey: parameters?.currentIssueKey,
                currentProjectId: parameters?.currentProjectId,
                showSubTasks: parameters?.showSubTasks,
                showSubTaskParent: parameters?.showSubTaskParent,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async matchIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/jql/match',
            method: 'POST',
            data: {
                issueIds: parameters.issueIds,
                jqls: parameters.jqls,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchForIssuesUsingJql(parameters, callback) {
        const config = {
            url: '/rest/api/3/search',
            method: 'GET',
            params: {
                jql: parameters.jql,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                validateQuery: parameters.validateQuery,
                fields: parameters.fields,
                expand: parameters.expand,
                properties: parameters.properties,
                fieldsByKeys: parameters.fieldsByKeys,
                failFast: parameters.failFast,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchForIssuesUsingJqlPost(parameters, callback) {
        const config = {
            url: '/rest/api/3/search',
            method: 'POST',
            data: {
                expand: parameters?.expand,
                fields: parameters?.fields,
                fieldsByKeys: parameters?.fieldsByKeys,
                jql: parameters?.jql,
                maxResults: parameters?.maxResults,
                properties: parameters?.properties,
                startAt: parameters?.startAt,
                validateQuery: parameters?.validateQuery,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async countIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/search/approximate-count',
            method: 'POST',
            data: {
                jql: parameters.jql,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchForIssuesIds(parameters, callback) {
        const config = {
            url: '/rest/api/3/search/id',
            method: 'POST',
            data: {
                jql: parameters.jql,
                maxResults: parameters.maxResults,
                nextPageToken: parameters.nextPageToken,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchForIssuesUsingJqlEnhancedSearch(parameters, callback) {
        const config = {
            url: '/rest/api/3/search/jql',
            method: 'GET',
            params: {
                jql: parameters.jql,
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
                fields: parameters.fields,
                expand: parameters.expand,
                properties: parameters.properties,
                fieldsByKeys: parameters.fieldsByKeys,
                failFast: parameters.failFast,
                reconcileIssues: parameters.reconcileIssues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchForIssuesUsingJqlEnhancedSearchPost(parameters, callback) {
        const config = {
            url: '/rest/api/3/search/jql',
            method: 'POST',
            data: {
                expand: parameters.expand,
                fields: parameters.fields,
                fieldsByKeys: parameters.fieldsByKeys,
                jql: parameters.jql,
                maxResults: parameters.maxResults,
                nextPageToken: parameters.nextPageToken,
                properties: parameters.properties,
                reconcileIssues: parameters.reconcileIssues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueSearch };
