// jira.js/dist/esm/version3/issueCustomFieldConfigurationApps.mjs
import { paramSerializer } from '../paramSerializer.mjs';

class IssueCustomFieldConfigurationApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCustomFieldsConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/3/app/field/context/configuration/list',
            method: 'POST',
            params: {
                id: parameters?.id,
                fieldContextId: paramSerializer('fieldContextId', parameters?.fieldContextId),
                issueId: parameters?.issueId,
                projectKeyOrId: parameters?.projectKeyOrId,
                issueTypeId: parameters?.issueTypeId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
            data: {
                fieldIdsOrKeys: parameters?.fieldIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCustomFieldConfiguration(parameters, callback) {
        const fieldIdOrKey = typeof parameters === 'string' ? parameters : parameters.fieldIdOrKey;
        const config = {
            url: `/rest/api/3/app/field/${fieldIdOrKey}/context/configuration`,
            method: 'GET',
            params: {
                id: typeof parameters !== 'string' && parameters.id,
                fieldContextId: typeof parameters !== 'string' && parameters.fieldContextId,
                issueId: typeof parameters !== 'string' && parameters.issueId,
                projectKeyOrId: typeof parameters !== 'string' && parameters.projectKeyOrId,
                issueTypeId: typeof parameters !== 'string' && parameters.issueTypeId,
                startAt: typeof parameters !== 'string' && parameters.startAt,
                maxResults: typeof parameters !== 'string' && parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/3/app/field/${parameters.fieldIdOrKey}/context/configuration`,
            method: 'PUT',
            data: {
                configurations: parameters.configurations,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueCustomFieldConfigurationApps };
