// jira.js/dist/esm/version3/issueProperties.mjs
class IssueProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async bulkSetIssuesProperties(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/properties',
            method: 'POST',
            data: {
                entitiesIds: parameters?.entitiesIds,
                properties: parameters?.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkSetIssuePropertiesByIssue(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/properties/multi',
            method: 'POST',
            data: {
                issues: parameters?.issues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkSetIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: {
                expression: parameters.expression,
                filter: parameters.filter,
                value: parameters.value,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkDeleteIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/properties/${parameters.propertyKey}`,
            method: 'DELETE',
            data: {
                currentValue: parameters.currentValue,
                entityIds: parameters.entityIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssuePropertyKeys(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueProperties };
