// jira.js/dist/esm/version3/issueFields.mjs
class IssueFields {
    client;
    constructor(client) {
        this.client = client;
    }
    async getFields(callback) {
        const config = {
            url: '/rest/api/3/field',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createCustomField(parameters, callback) {
        const config = {
            url: '/rest/api/3/field',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
                searcherKey: parameters.searcherKey,
                type: parameters.type,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldsPaginated(parameters, callback) {
        const config = {
            url: '/rest/api/3/field/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                type: parameters?.type,
                id: parameters?.id,
                query: parameters?.query,
                orderBy: parameters?.orderBy,
                expand: parameters?.expand,
                projectIds: parameters?.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getTrashedFieldsPaginated(parameters, callback) {
        const config = {
            url: '/rest/api/3/field/search/trashed',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                query: parameters?.query,
                expand: parameters?.expand,
                orderBy: parameters?.orderBy,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomField(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
                searcherKey: parameters.searcherKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteCustomField(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async restoreCustomField(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.id}/restore`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async trashCustomField(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.id}/trash`,
            method: 'POST',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectFields(parameters, callback) {
        const config = {
            url: '/rest/api/3/projects/fields',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
                workTypeId: parameters.workTypeId,
                fieldId: parameters.fieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueFields };
