// jira.js/dist/esm/version3/issueTypeScreenSchemes.mjs
class IssueTypeScreenSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueTypeScreenSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuetypescreenscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                queryString: parameters?.queryString,
                orderBy: parameters?.orderBy,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuetypescreenscheme',
            method: 'POST',
            data: {
                description: parameters.description,
                issueTypeMappings: parameters.issueTypeMappings,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeScreenSchemeMappings(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuetypescreenscheme/mapping',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                issueTypeScreenSchemeId: parameters?.issueTypeScreenSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueTypeScreenSchemeProjectAssociations(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuetypescreenscheme/project',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignIssueTypeScreenSchemeToProject(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuetypescreenscheme/project',
            method: 'PUT',
            data: {
                issueTypeScreenSchemeId: parameters?.issueTypeScreenSchemeId,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async appendMappingsForIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}/mapping`,
            method: 'PUT',
            data: {
                issueTypeMappings: parameters.issueTypeMappings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDefaultScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}/mapping/default`,
            method: 'PUT',
            data: {
                screenSchemeId: parameters.screenSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeMappingsFromIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}/mapping/remove`,
            method: 'POST',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectsForIssueTypeScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuetypescreenscheme/${parameters.issueTypeScreenSchemeId}/project`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                query: parameters.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { IssueTypeScreenSchemes };
