// jira.js/dist/esm/version3/jqlFunctionsApps.mjs
class JqlFunctionsApps {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPrecomputations(parameters, callback) {
        const config = {
            url: '/rest/api/3/jql/function/computation',
            method: 'GET',
            params: {
                functionKey: parameters?.functionKey,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                orderBy: parameters?.orderBy,
                filter: parameters?.filter,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePrecomputations(parameters, callback) {
        const config = {
            url: '/rest/api/3/jql/function/computation',
            method: 'POST',
            params: {
                skipNotFoundPrecomputations: parameters.skipNotFoundPrecomputations,
            },
            data: {
                values: parameters.values,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPrecomputationsByID(parameters, callback) {
        const config = {
            url: '/rest/api/3/jql/function/computation/search',
            method: 'POST',
            params: {
                orderBy: parameters.orderBy,
            },
            data: {
                precomputationIDs: parameters.precomputationIDs,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { JqlFunctionsApps };
