// jira.js/dist/esm/version3/myself.mjs
class Myself {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPreference(parameters, callback) {
        const config = {
            url: '/rest/api/3/mypreferences',
            method: 'GET',
            params: {
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setPreference(parameters, callback) {
        const config = {
            url: '/rest/api/3/mypreferences',
            method: 'PUT',
            params: {
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removePreference(parameters, callback) {
        const config = {
            url: '/rest/api/3/mypreferences',
            method: 'DELETE',
            params: {
                key: parameters.key,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getLocale(callback) {
        const config = {
            url: '/rest/api/3/mypreferences/locale',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getCurrentUser(parameters, callback) {
        const config = {
            url: '/rest/api/3/myself',
            method: 'GET',
            params: {
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { Myself };
