// jira.js/dist/esm/version3/projectRoleActors.mjs
class ProjectRoleActors {
    client;
    constructor(client) {
        this.client = client;
    }
    async addActorUsers(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/role/${parameters.id}`,
            method: 'POST',
            data: {
                group: parameters.group,
                groupId: parameters.groupId,
                user: parameters.user,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setActors(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/role/${parameters.id}`,
            method: 'PUT',
            data: {
                categorisedActors: parameters.categorisedActors,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteActor(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/role/${parameters.id}`,
            method: 'DELETE',
            params: {
                user: parameters.user,
                group: parameters.group,
                groupId: parameters.groupId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectRoleActorsForRole(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/role/${id}/actors`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addProjectRoleActorsToRole(parameters, callback) {
        const config = {
            url: `/rest/api/3/role/${parameters.id}/actors`,
            method: 'POST',
            data: {
                group: parameters.group,
                groupId: parameters.groupId,
                user: parameters.user,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProjectRoleActorsFromRole(parameters, callback) {
        const config = {
            url: `/rest/api/3/role/${parameters.id}/actors`,
            method: 'DELETE',
            params: {
                user: parameters.user,
                groupId: parameters.groupId,
                group: parameters.group,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { ProjectRoleActors };
