// jira.js/dist/esm/version3/projectComponents.mjs
class ProjectComponents {
    client;
    constructor(client) {
        this.client = client;
    }
    async findComponentsForProjects(parameters, callback) {
        const config = {
            url: '/rest/api/3/component',
            method: 'GET',
            params: {
                projectIdsOrKeys: parameters.projectIdsOrKeys,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                orderBy: parameters.orderBy,
                query: parameters.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createComponent(parameters, callback) {
        const config = {
            url: '/rest/api/3/component',
            method: 'POST',
            data: {
                ari: parameters.ari,
                assignee: parameters.assignee,
                assigneeType: parameters.assigneeType,
                description: parameters.description,
                id: parameters.id,
                isAssigneeTypeValid: parameters.isAssigneeTypeValid,
                lead: parameters.lead,
                leadAccountId: parameters.leadAccountId,
                leadUserName: parameters.leadUserName,
                metadata: parameters.metadata,
                name: parameters.name,
                project: parameters.project,
                projectId: parameters.projectId,
                realAssignee: parameters.realAssignee,
                realAssigneeType: parameters.realAssigneeType,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getComponent(parameters, callback) {
        const config = {
            url: `/rest/api/3/component/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateComponent(parameters, callback) {
        const config = {
            url: `/rest/api/3/component/${parameters.id}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
                leadUserName: parameters.leadUserName,
                leadAccountId: parameters.leadAccountId,
                assigneeType: parameters.assigneeType,
                project: parameters.project,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteComponent(parameters, callback) {
        const config = {
            url: `/rest/api/3/component/${parameters.id}`,
            method: 'DELETE',
            params: {
                moveIssuesTo: parameters.moveIssuesTo,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getComponentRelatedIssues(parameters, callback) {
        const config = {
            url: `/rest/api/3/component/${parameters.id}/relatedIssueCounts`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectComponentsPaginated(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/component`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                orderBy: parameters.orderBy,
                componentSource: parameters.componentSource,
                query: parameters.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectComponents(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/components`,
            method: 'GET',
            params: {
                componentSource: parameters.componentSource,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { ProjectComponents };
