// jira.js/dist/esm/version3/projectProperties.mjs
class ProjectProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getProjectPropertyKeys(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/3/project/${projectIdOrKey}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setProjectProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'PUT',
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProjectProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { ProjectProperties };
