// jira.js/dist/esm/version3/screenTabFields.mjs
class ScreenTabFields {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllScreenTabFields(parameters, callback) {
        const config = {
            url: `/rest/api/3/screens/${parameters.screenId}/tabs/${parameters.tabId}/fields`,
            method: 'GET',
            params: {
                projectKey: parameters.projectKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addScreenTabField(parameters, callback) {
        const config = {
            url: `/rest/api/3/screens/${parameters.screenId}/tabs/${parameters.tabId}/fields`,
            method: 'POST',
            data: {
                fieldId: parameters.fieldId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeScreenTabField(parameters, callback) {
        const config = {
            url: `/rest/api/3/screens/${parameters.screenId}/tabs/${parameters.tabId}/fields/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async moveScreenTabField(parameters, callback) {
        const config = {
            url: `/rest/api/3/screens/${parameters.screenId}/tabs/${parameters.tabId}/fields/${parameters.id}/move`,
            method: 'POST',
            data: {
                after: parameters.after,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { ScreenTabFields };
