// jira.js/dist/esm/version3/teamsInPlan.mjs
class TeamsInPlan {
    client;
    constructor(client) {
        this.client = client;
    }
    async getTeams(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team`,
            method: 'GET',
            params: {
                cursor: parameters.cursor,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addAtlassianTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/atlassian`,
            method: 'POST',
            data: {
                capacity: parameters.capacity,
                id: parameters.id,
                issueSourceId: parameters.issueSourceId,
                planningStyle: parameters.planningStyle,
                sprintLength: parameters.sprintLength,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAtlassianTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/atlassian/${parameters.atlassianTeamId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateAtlassianTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/atlassian/${parameters.atlassianTeamId}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async removeAtlassianTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/atlassian/${parameters.atlassianTeamId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async createPlanOnlyTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/planonly`,
            method: 'POST',
            data: {
                capacity: parameters.capacity,
                issueSourceId: parameters.issueSourceId,
                memberAccountIds: parameters.memberAccountIds,
                name: parameters.name,
                planningStyle: parameters.planningStyle,
                sprintLength: parameters.sprintLength,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPlanOnlyTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/planonly/${parameters.planOnlyTeamId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePlanOnlyTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/planonly/${parameters.planOnlyTeamId}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deletePlanOnlyTeam(parameters, callback) {
        const config = {
            url: `/rest/api/3/plans/plan/${parameters.planId}/team/planonly/${parameters.planOnlyTeamId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

export { TeamsInPlan };
