// jira.js/dist/esm/version3/userSearch.mjs
import { paramSerializer } from '../paramSerializer.mjs';

class UserSearch {
    client;
    constructor(client) {
        this.client = client;
    }
    async findBulkAssignableUsers(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/assignable/multiProjectSearch',
            method: 'GET',
            params: {
                query: parameters.query,
                username: parameters.username,
                accountId: parameters.accountId,
                projectKeys: parameters.projectKeys,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findAssignableUsers(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/assignable/search',
            method: 'GET',
            params: {
                query: parameters?.query,
                sessionId: parameters?.sessionId,
                username: parameters?.username,
                accountId: parameters?.accountId,
                project: parameters?.project,
                issueKey: parameters?.issueKey,
                issueId: parameters?.issueId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                actionDescriptorId: parameters?.actionDescriptorId,
                recommend: parameters?.recommend,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUsersWithAllPermissions(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/permission/search',
            method: 'GET',
            params: {
                query: parameters.query,
                username: parameters.username,
                accountId: parameters.accountId,
                permissions: parameters.permissions,
                issueKey: parameters.issueKey,
                projectKey: parameters.projectKey,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUsersForPicker(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/picker',
            method: 'GET',
            params: {
                query: parameters.query,
                maxResults: parameters.maxResults,
                showAvatar: parameters.showAvatar,
                excludeAccountIds: paramSerializer('excludeAccountIds', parameters.excludeAccountIds),
                avatarSize: parameters.avatarSize,
                excludeConnectUsers: parameters.excludeConnectUsers,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUsers(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/search',
            method: 'GET',
            params: {
                query: parameters?.query,
                username: parameters?.username,
                accountId: parameters?.accountId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                property: parameters?.property,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUsersByQuery(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/search/query',
            method: 'GET',
            params: {
                query: parameters.query,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUserKeysByQuery(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/search/query/key',
            method: 'GET',
            params: {
                query: parameters.query,
                startAt: parameters.startAt,
                maxResult: parameters.maxResult || parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async findUsersWithBrowsePermission(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/viewissue/search',
            method: 'GET',
            params: {
                query: parameters?.query,
                username: parameters?.username,
                accountId: parameters?.accountId,
                issueKey: parameters?.issueKey,
                projectKey: parameters?.projectKey,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { UserSearch };
