// jira.js/dist/esm/version3/workflowSchemes.mjs
class WorkflowSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllWorkflowSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createWorkflowScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme',
            method: 'POST',
            data: {
                defaultWorkflow: parameters.defaultWorkflow,
                description: parameters.description,
                draft: parameters.draft,
                id: parameters.id,
                issueTypeMappings: parameters.issueTypeMappings,
                issueTypes: parameters.issueTypes,
                lastModified: parameters.lastModified,
                lastModifiedUser: parameters.lastModifiedUser,
                name: parameters.name,
                originalDefaultWorkflow: parameters.originalDefaultWorkflow,
                originalIssueTypeMappings: parameters.originalIssueTypeMappings,
                self: parameters.self,
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async switchWorkflowSchemeForProject(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/project/switch',
            method: 'POST',
            data: {
                mappingsByIssueTypeOverride: parameters.mappingsByIssueTypeOverride,
                projectId: parameters.projectId,
                targetSchemeId: parameters.targetSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async readWorkflowSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/read',
            method: 'POST',
            data: {
                projectIds: parameters.projectIds,
                workflowSchemeIds: parameters.workflowSchemeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/update',
            method: 'POST',
            data: {
                defaultWorkflowId: parameters.defaultWorkflowId,
                description: parameters.description,
                id: parameters.id,
                name: parameters.name,
                statusMappingsByIssueTypeOverride: parameters.statusMappingsByIssueTypeOverride,
                statusMappingsByWorkflows: parameters.statusMappingsByWorkflows,
                version: parameters.version,
                workflowsForIssueTypes: parameters.workflowsForIssueTypes,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowSchemeMappings(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/update/mappings',
            method: 'POST',
            data: {
                defaultWorkflowId: parameters.defaultWorkflowId,
                id: parameters.id,
                workflowsForIssueTypes: parameters.workflowsForIssueTypes,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}`,
            method: 'GET',
            params: {
                returnDraftIfExists: typeof parameters !== 'string' && parameters.returnDraftIfExists,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
                defaultWorkflow: parameters.defaultWorkflow,
                issueTypeMappings: parameters.issueTypeMappings,
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowScheme(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getDefaultWorkflow(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}/default`,
            method: 'GET',
            params: {
                returnDraftIfExists: typeof parameters !== 'string' && parameters.returnDraftIfExists,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDefaultWorkflow(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}/default`,
            method: 'PUT',
            data: {
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                workflow: parameters.workflow,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteDefaultWorkflow(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}/default`,
            method: 'DELETE',
            params: {
                updateDraftIfNeeded: typeof parameters !== 'string' && parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowSchemeIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
            method: 'GET',
            params: {
                returnDraftIfExists: parameters.returnDraftIfExists,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setWorkflowSchemeIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
            method: 'PUT',
            data: parameters.details,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowSchemeIssueType(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
            method: 'DELETE',
            params: {
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflow(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}/workflow`,
            method: 'GET',
            params: {
                workflowName: typeof parameters !== 'string' && parameters.workflowName,
                returnDraftIfExists: typeof parameters !== 'string' && parameters.returnDraftIfExists,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowMapping(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.id}/workflow`,
            method: 'PUT',
            params: {
                workflowName: parameters.workflowName,
            },
            data: {
                defaultMapping: parameters.defaultMapping,
                issueTypes: parameters.issueTypes,
                updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                workflow: parameters.workflow,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowMapping(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/workflowscheme/${id}/workflow`,
            method: 'DELETE',
            params: {
                workflowName: typeof parameters !== 'string' && parameters.workflowName,
                updateDraftIfNeeded: typeof parameters !== 'string' && parameters.updateDraftIfNeeded,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectUsagesForWorkflowScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/workflowscheme/${parameters.workflowSchemeId}/projectUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

export { WorkflowSchemes };
