// jira.js/dist/esm/config.mjs
import { z } from 'zod';

// Authentication schemas
// const JWTSchema = z.object({
//   issuer: z.string(),
//   secret: z.string(),
//   expiryTimeSeconds: z.number().optional()
// });
const BasicAuthSchema = z
    .object({
    email: z.string(),
    apiToken: z.string(),
})
    .strict();
const OAuth2Schema = z
    .object({
    accessToken: z.string(),
})
    .strict();
// Middlewares schemas
const MiddlewaresSchema = z
    .object({
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    onError: z.optional(z.any()),
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    onResponse: z.optional(z.any()),
})
    .strict();
const ConfigSchema = z
    .object({
    host: z.string().url(),
    strictGDPR: z.boolean().optional(),
    /** Adds `'X-Atlassian-Token': 'no-check'` to each request header */
    noCheckAtlassianToken: z.boolean().optional(),
    baseRequestConfig: z.any().optional(),
    authentication: z.union([z.object({ basic: BasicAuthSchema }), z.object({ oauth2: OAuth2Schema })]).optional(),
    middlewares: MiddlewaresSchema.optional(),
})
    .strict();

export { BasicAuthSchema, ConfigSchema, MiddlewaresSchema, OAuth2Schema };
