// jira.js/dist/esm/createClient.mjs
import { AgileClient } from './agile/client/agileClient.mjs';
import { BaseClient } from './clients/baseClient.mjs';
import 'mime';
import { Version2Client } from './version2/client/version2Client.mjs';
import 'zod';
import './version3/models/projectField.mjs';
import './version3/models/jiraStatus.mjs';
import './version3/models/projectId.mjs';
import './version3/models/projectIssueTypes.mjs';
import './version3/models/statusScope.mjs';
import { Version3Client } from './version3/client/version3Client.mjs';
import './version3/parameters/getStatusesByName.mjs';
import { ServiceDeskClient } from './serviceDesk/client/serviceDeskClient.mjs';

var ClientType;
(function (ClientType) {
    ClientType["Agile"] = "agile";
    ClientType["Version2"] = "version2";
    ClientType["Version3"] = "version3";
    ClientType["ServiceDesk"] = "serviceDesk";
})(ClientType || (ClientType = {}));
function createClient(clientType, config) {
    switch (clientType) {
        case ClientType.Agile:
            return new AgileClient(config);
        case ClientType.Version2:
            return new Version2Client(config);
        case ClientType.Version3:
            return new Version3Client(config);
        case ClientType.ServiceDesk:
            return new ServiceDeskClient(config);
        default:
            return new BaseClient(config);
    }
}

export { ClientType, createClient };
