// jira.js/dist/cjs/version3/filterSharing.cjs
'use strict';

class FilterSharing {
    client;
    constructor(client) {
        this.client = client;
    }
    async getDefaultShareScope(callback) {
        const config = {
            url: '/rest/api/3/filter/defaultShareScope',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultShareScope(parameters, callback) {
        const scope = typeof parameters === 'string' ? parameters : parameters.scope;
        const config = {
            url: '/rest/api/3/filter/defaultShareScope',
            method: 'PUT',
            data: {
                scope,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSharePermissions(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/filter/${id}/permission`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/3/filter/${parameters.id}/permission`,
            method: 'POST',
            data: {
                accountId: parameters.accountId,
                groupId: parameters.groupId,
                groupname: parameters.groupname,
                projectId: parameters.projectId,
                projectRoleId: parameters.projectRoleId,
                rights: parameters.rights,
                type: parameters.type,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/3/filter/${parameters.id}/permission/${parameters.permissionId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteSharePermission(parameters, callback) {
        const config = {
            url: `/rest/api/3/filter/${parameters.id}/permission/${parameters.permissionId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.FilterSharing = FilterSharing;
