// jira.js/dist/cjs/version3/issueWorklogProperties.cjs
'use strict';

class IssueWorklogProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getWorklogPropertyKeys(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.worklogId}/properties`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorklogProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.worklogId}/properties/${parameters.propertyKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setWorklogProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.worklogId}/properties/${parameters.propertyKey}`,
            method: 'PUT',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorklogProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.worklogId}/properties/${parameters.propertyKey}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueWorklogProperties = IssueWorklogProperties;
