// jira.js/dist/cjs/version3/issueSecuritySchemes.cjs
'use strict';

class IssueSecuritySchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueSecuritySchemes(callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssueSecurityScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes',
            method: 'POST',
            data: {
                description: parameters.description,
                levels: parameters.levels,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSecurityLevels(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/level',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                schemeId: parameters?.schemeId,
                onlyDefault: parameters?.onlyDefault,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultLevels(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/level/default',
            method: 'PUT',
            data: {
                defaultValues: parameters?.defaultValues,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getSecurityLevelMembers(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/level/member',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                schemeId: parameters?.schemeId,
                levelId: parameters?.levelId,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchProjectsUsingSecuritySchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/project',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                issueSecuritySchemeId: parameters?.issueSecuritySchemeId,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async associateSchemesToProjects(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/project',
            method: 'PUT',
            data: {
                oldToNewSecurityLevelMappings: parameters.oldToNewSecurityLevelMappings,
                projectId: parameters.projectId,
                schemeId: parameters.schemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchSecuritySchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/issuesecurityschemes/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueSecurityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateIssueSecurityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteSecurityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async addSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}/level`,
            method: 'PUT',
            data: {
                levels: parameters.levels,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeLevel(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}`,
            method: 'DELETE',
            params: {
                replaceWith: parameters.replaceWith,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addSecurityLevelMembers(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}/member`,
            method: 'PUT',
            data: {
                members: parameters.members,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeMemberFromSecurityLevel(parameters, callback) {
        const config = {
            url: `/rest/api/3/issuesecurityschemes/${parameters.schemeId}/level/${parameters.levelId}/member/${parameters.memberId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueSecuritySchemes = IssueSecuritySchemes;
