// jira.js/dist/cjs/version3/issueWatchers.cjs
'use strict';

class IssueWatchers {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIsWatchingIssueBulk(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/watching',
            method: 'POST',
            data: {
                issueIds: parameters?.issueIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueWatchers(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/watchers`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async addWatcher(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/watchers`,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            data: parameters.accountId,
        };
        return this.client.sendRequest(config, callback);
    }
    async removeWatcher(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/watchers`,
            method: 'DELETE',
            params: {
                username: parameters.username,
                accountId: parameters.accountId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueWatchers = IssueWatchers;
