// jira.js/dist/cjs/version3/issueWorklogs.cjs
'use strict';

class IssueWorklogs {
    client;
    constructor(client) {
        this.client = client;
    }
    async getIssueWorklog(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                startedAfter: parameters.startedAfter,
                startedBefore: parameters.startedBefore,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async addWorklog(parameters, callback) {
        let comment;
        if (typeof parameters.comment === 'string') {
            comment = {
                type: 'doc',
                version: 1,
                content: [
                    {
                        type: 'paragraph',
                        content: [
                            {
                                type: 'text',
                                text: parameters.comment,
                            },
                        ],
                    },
                ],
            };
        }
        else {
            comment = parameters.comment;
        }
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog`,
            method: 'POST',
            params: {
                notifyUsers: parameters.notifyUsers,
                adjustEstimate: parameters.adjustEstimate,
                newEstimate: parameters.newEstimate,
                reduceBy: parameters.reduceBy,
                expand: parameters.expand,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
            data: {
                author: parameters.author,
                comment,
                created: parameters.created,
                id: parameters.id,
                issueId: parameters.issueId,
                properties: parameters.properties,
                self: parameters.self,
                started: parameters.started,
                timeSpent: parameters.timeSpent,
                timeSpentSeconds: parameters.timeSpentSeconds,
                updateAuthor: parameters.updateAuthor,
                updated: parameters.updated,
                visibility: parameters.visibility,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkDeleteWorklogs(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog`,
            method: 'DELETE',
            params: {
                adjustEstimate: parameters.adjustEstimate,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
            data: {
                ids: parameters.ids,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkMoveWorklogs(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/move`,
            method: 'POST',
            params: {
                adjustEstimate: parameters.adjustEstimate,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
            data: parameters.worklogs,
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorklog(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.id}`,
            method: 'GET',
            params: {
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorklog(parameters, callback) {
        let comment;
        if (typeof parameters.comment === 'string') {
            comment = {
                type: 'doc',
                version: 1,
                content: [
                    {
                        type: 'paragraph',
                        content: [
                            {
                                type: 'text',
                                text: parameters.comment,
                            },
                        ],
                    },
                ],
            };
        }
        else {
            comment = parameters.comment;
        }
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.id}`,
            method: 'PUT',
            params: {
                notifyUsers: parameters.notifyUsers,
                adjustEstimate: parameters.adjustEstimate,
                newEstimate: parameters.newEstimate,
                expand: parameters.expand,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
            data: {
                comment,
                visibility: parameters.visibility,
                started: parameters.started,
                timeSpent: parameters.timeSpent,
                timeSpentSeconds: parameters.timeSpentSeconds,
                properties: parameters.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorklog(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/worklog/${parameters.id}`,
            method: 'DELETE',
            params: {
                notifyUsers: parameters.notifyUsers,
                adjustEstimate: parameters.adjustEstimate,
                newEstimate: parameters.newEstimate,
                increaseBy: parameters.increaseBy,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIdsOfWorklogsDeletedSince(parameters, callback) {
        const config = {
            url: '/rest/api/3/worklog/deleted',
            method: 'GET',
            params: {
                since: parameters?.since,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorklogsForIds(parameters, callback) {
        const config = {
            url: '/rest/api/3/worklog/list',
            method: 'POST',
            params: {
                expand: parameters?.expand,
            },
            data: {
                ids: parameters?.ids,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIdsOfWorklogsModifiedSince(parameters, callback) {
        const config = {
            url: '/rest/api/3/worklog/updated',
            method: 'GET',
            params: {
                since: parameters?.since,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueWorklogs = IssueWorklogs;
