// jira.js/dist/cjs/version3/issueFieldConfigurations.cjs
'use strict';

class IssueFieldConfigurations {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllFieldConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfiguration',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                isDefault: parameters?.isDefault,
                query: parameters?.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createFieldConfiguration(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfiguration',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfiguration/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFieldConfiguration(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfiguration/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationItems(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfiguration/${parameters.id}/fields`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfigurationItems(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfiguration/${parameters.id}/fields`,
            method: 'PUT',
            data: {
                fieldConfigurationItems: parameters.fieldConfigurationItems,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllFieldConfigurationSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfigurationscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfigurationscheme',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationSchemeMappings(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfigurationscheme/mapping',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                fieldConfigurationSchemeId: parameters?.fieldConfigurationSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getFieldConfigurationSchemeProjectMapping(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfigurationscheme/project',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: parameters.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignFieldConfigurationSchemeToProject(parameters, callback) {
        const config = {
            url: '/rest/api/3/fieldconfigurationscheme/project',
            method: 'PUT',
            data: {
                fieldConfigurationSchemeId: parameters?.fieldConfigurationSchemeId,
                projectId: parameters?.projectId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfigurationscheme/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfigurationscheme/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async setFieldConfigurationSchemeMapping(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfigurationscheme/${parameters.id}/mapping`,
            method: 'PUT',
            data: {
                mappings: parameters.mappings,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeIssueTypesFromGlobalFieldConfigurationScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/fieldconfigurationscheme/${parameters.id}/mapping/delete`,
            method: 'POST',
            data: {
                issueTypeIds: parameters.issueTypeIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueFieldConfigurations = IssueFieldConfigurations;
