// jira.js/dist/cjs/version3/issueLinks.cjs
'use strict';

class IssueLinks {
    client;
    constructor(client) {
        this.client = client;
    }
    async linkIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issueLink',
            method: 'POST',
            data: {
                comment: parameters.comment,
                inwardIssue: parameters.inwardIssue,
                outwardIssue: parameters.outwardIssue,
                type: parameters.type,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueLink(parameters, callback) {
        const config = {
            url: `/rest/api/3/issueLink/${parameters.linkId}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssueLink(parameters, callback) {
        const config = {
            url: `/rest/api/3/issueLink/${parameters.linkId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueLinks = IssueLinks;
