// jira.js/dist/cjs/version3/issuePriorities.cjs
'use strict';

var paramSerializer = require('../paramSerializer.cjs');

class IssuePriorities {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPriorities(callback) {
        const config = {
            url: '/rest/api/3/priority',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createPriority(parameters, callback) {
        const config = {
            url: '/rest/api/3/priority',
            method: 'POST',
            data: {
                avatarId: parameters.avatarId,
                description: parameters.description,
                iconUrl: parameters.iconUrl,
                name: parameters.name,
                statusColor: parameters.statusColor,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultPriority(parameters, callback) {
        const config = {
            url: '/rest/api/3/priority/default',
            method: 'PUT',
            data: {
                id: parameters?.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async movePriorities(parameters, callback) {
        const config = {
            url: '/rest/api/3/priority/move',
            method: 'PUT',
            data: {
                after: parameters.after,
                ids: parameters.ids,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchPriorities(parameters, callback) {
        const config = {
            url: '/rest/api/3/priority/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                projectId: paramSerializer.paramSerializer('projectId', parameters?.projectId),
                priorityName: parameters?.priorityName,
                onlyDefault: parameters?.onlyDefault,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPriority(parameters, callback) {
        const config = {
            url: `/rest/api/3/priority/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePriority(parameters, callback) {
        const config = {
            url: `/rest/api/3/priority/${parameters.id}`,
            method: 'PUT',
            data: {
                avatarId: parameters.avatarId,
                description: parameters.description,
                iconUrl: parameters.iconUrl,
                name: parameters.name,
                statusColor: parameters.statusColor,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deletePriority(parameters, callback) {
        const config = {
            url: `/rest/api/3/priority/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssuePriorities = IssuePriorities;
