// jira.js/dist/cjs/version3/issueCustomFieldOptions.cjs
'use strict';

class IssueCustomFieldOptions {
    client;
    constructor(client) {
        this.client = client;
    }
    async getCustomFieldOption(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/customFieldOption/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getOptionsForContext(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option`,
            method: 'GET',
            params: {
                optionId: parameters.optionId,
                onlyOptions: parameters.onlyOptions,
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createCustomFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option`,
            method: 'POST',
            data: {
                options: parameters.options,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateCustomFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option`,
            method: 'PUT',
            data: {
                options: parameters.options,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async reorderCustomFieldOptions(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option/move`,
            method: 'PUT',
            data: {
                after: parameters.after,
                customFieldOptionIds: parameters.customFieldOptionIds,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteCustomFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option/${parameters.optionId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async replaceCustomFieldOption(parameters, callback) {
        const config = {
            url: `/rest/api/3/field/${parameters.fieldId}/context/${parameters.contextId}/option/${parameters.optionId}/issue`,
            method: 'DELETE',
            params: {
                replaceWith: parameters.replaceWith,
                jql: parameters.jql,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueCustomFieldOptions = IssueCustomFieldOptions;
