// jira.js/dist/cjs/version3/issueResolutions.cjs
'use strict';

class IssueResolutions {
    client;
    constructor(client) {
        this.client = client;
    }
    async getResolutions(callback) {
        const config = {
            url: '/rest/api/3/resolution',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createResolution(parameters, callback) {
        const config = {
            url: '/rest/api/3/resolution',
            method: 'POST',
            data: parameters,
        };
        return this.client.sendRequest(config, callback);
    }
    async setDefaultResolution(parameters, callback) {
        const config = {
            url: '/rest/api/3/resolution/default',
            method: 'PUT',
            data: {
                id: parameters.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async moveResolutions(parameters, callback) {
        const config = {
            url: '/rest/api/3/resolution/move',
            method: 'PUT',
            data: {
                after: parameters.after,
                ids: parameters.ids,
                position: parameters.position,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async searchResolutions(parameters, callback) {
        const config = {
            url: '/rest/api/3/resolution/search',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                onlyDefault: parameters?.onlyDefault,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getResolution(parameters, callback) {
        const config = {
            url: `/rest/api/3/resolution/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateResolution(parameters, callback) {
        const config = {
            url: `/rest/api/3/resolution/${parameters.id}`,
            method: 'PUT',
            data: {
                ...parameters,
                name: parameters.name,
                description: parameters.description,
                id: undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteResolution(parameters, callback) {
        const config = {
            url: `/rest/api/3/resolution/${parameters.id}`,
            method: 'DELETE',
            params: {
                replaceWith: parameters.replaceWith,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.IssueResolutions = IssueResolutions;
