// jira.js/dist/cjs/version3/issues.cjs
'use strict';

class Issues {
    client;
    constructor(client) {
        this.client = client;
    }
    async getBulkChangelogs(parameters, callback) {
        const config = {
            url: '/rest/api/3/changelog/bulkfetch',
            method: 'POST',
            data: {
                fieldIds: parameters.fieldIds,
                issueIdsOrKeys: parameters.issueIdsOrKeys,
                maxResults: parameters.maxResults,
                nextPageToken: parameters.nextPageToken,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getEvents(callback) {
        const config = {
            url: '/rest/api/3/events',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssue(parameters, callback) {
        if (typeof parameters.fields.description === 'string') {
            parameters.fields.description = {
                type: 'doc',
                version: 1,
                content: [
                    {
                        type: 'paragraph',
                        content: [
                            {
                                text: parameters.fields.description,
                                type: 'text',
                            },
                        ],
                    },
                ],
            };
        }
        const config = {
            url: '/rest/api/3/issue',
            method: 'POST',
            params: {
                updateHistory: parameters.updateHistory,
            },
            data: {
                fields: parameters.fields,
                historyMetadata: parameters.historyMetadata,
                properties: parameters.properties,
                transition: parameters.transition,
                update: parameters.update,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async archiveIssuesAsync(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/archive',
            method: 'POST',
            data: {
                jql: parameters.jql,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async archiveIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/archive',
            method: 'PUT',
            data: {
                issueIdsOrKeys: parameters.issueIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/bulk',
            method: 'POST',
            data: {
                issueUpdates: parameters?.issueUpdates,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async bulkFetchIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/bulkfetch',
            method: 'POST',
            data: {
                expand: parameters.expand,
                fields: parameters.fields,
                fieldsByKeys: parameters.fieldsByKeys,
                issueIdsOrKeys: parameters.issueIdsOrKeys,
                properties: parameters.properties,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCreateIssueMeta(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/createmeta',
            method: 'GET',
            params: {
                projectIds: parameters?.projectIds,
                projectKeys: parameters?.projectKeys,
                issuetypeIds: parameters?.issuetypeIds,
                issuetypeNames: parameters?.issuetypeNames,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCreateIssueMetaIssueTypes(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/createmeta/${parameters.projectIdOrKey}/issuetypes`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getCreateIssueMetaIssueTypeId(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/createmeta/${parameters.projectIdOrKey}/issuetypes/${parameters.issueTypeId}`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssueLimitReport(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/limit/report',
            method: 'GET',
            params: {
                isReturningKeys: parameters?.isReturningKeys,
            },
            data: {
                issuesApproachingLimitParams: parameters?.issuesApproachingLimitParams,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async unarchiveIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issue/unarchive',
            method: 'PUT',
            data: {
                issueIdsOrKeys: parameters.issueIdsOrKeys,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getIssue(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}`,
            method: 'GET',
            params: {
                fields: parameters.fields,
                fieldsByKeys: parameters.fieldsByKeys,
                expand: parameters.expand,
                properties: parameters.properties,
                updateHistory: parameters.updateHistory,
                failFast: parameters.failFast,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async editIssue(parameters, callback) {
        if (parameters.fields?.description && typeof parameters.fields.description === 'string') {
            const { fields: { description }, } = await this.getIssue({ issueIdOrKey: parameters.issueIdOrKey });
            parameters.fields.description = {
                type: 'doc',
                version: description?.version ?? 1,
                content: [
                    {
                        type: 'paragraph',
                        content: [
                            {
                                text: parameters.fields.description,
                                type: 'text',
                            },
                        ],
                    },
                ],
            };
        }
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}`,
            method: 'PUT',
            params: {
                notifyUsers: parameters.notifyUsers,
                overrideScreenSecurity: parameters.overrideScreenSecurity,
                overrideEditableFlag: parameters.overrideEditableFlag,
                returnIssue: parameters.returnIssue,
                expand: parameters.expand,
            },
            data: {
                fields: parameters.fields,
                historyMetadata: parameters.historyMetadata,
                properties: parameters.properties,
                transition: parameters.transition,
                update: parameters.update,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteIssue(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}`,
            method: 'DELETE',
            params: {
                deleteSubtasks: parameters.deleteSubtasks,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignIssue(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/assignee`,
            method: 'PUT',
            data: {
                accountId: parameters.accountId,
                accountType: parameters.accountType,
                active: parameters.active,
                appType: parameters.appType,
                applicationRoles: parameters.applicationRoles,
                avatarUrls: parameters.avatarUrls,
                displayName: parameters.displayName,
                emailAddress: parameters.emailAddress,
                expand: parameters.expand,
                groups: parameters.groups,
                key: parameters.key,
                locale: parameters.locale,
                name: parameters.name,
                self: parameters.self,
                timeZone: parameters.timeZone,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getChangeLogs(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/changelog`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getChangeLogsByIds(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/changelog/list`,
            method: 'POST',
            data: {
                changelogIds: parameters.changelogIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getEditIssueMeta(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/editmeta`,
            method: 'GET',
            params: {
                overrideScreenSecurity: parameters.overrideScreenSecurity,
                overrideEditableFlag: parameters.overrideEditableFlag,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async notify(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/notify`,
            method: 'POST',
            data: {
                htmlBody: parameters.htmlBody,
                restrict: parameters.restrict,
                subject: parameters.subject,
                textBody: parameters.textBody,
                to: parameters.to,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getTransitions(parameters, callback) {
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/transitions`,
            method: 'GET',
            params: {
                expand: parameters.expand,
                transitionId: parameters.transitionId,
                skipRemoteOnlyCondition: parameters.skipRemoteOnlyCondition,
                includeUnavailableTransitions: parameters.includeUnavailableTransitions,
                sortByOpsBarAndStatus: parameters.sortByOpsBarAndStatus,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async doTransition(parameters, callback) {
        if (parameters.fields?.description && typeof parameters.fields.description === 'string') {
            parameters.fields.description = {
                type: 'doc',
                version: 1,
                content: [
                    {
                        type: 'paragraph',
                        content: [
                            {
                                text: parameters.fields.description,
                                type: 'text',
                            },
                        ],
                    },
                ],
            };
        }
        const config = {
            url: `/rest/api/3/issue/${parameters.issueIdOrKey}/transitions`,
            method: 'POST',
            data: {
                fields: parameters.fields,
                historyMetadata: parameters.historyMetadata,
                properties: parameters.properties,
                transition: parameters.transition,
                update: parameters.update,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async exportArchivedIssues(parameters, callback) {
        const config = {
            url: '/rest/api/3/issues/archive/export',
            method: 'PUT',
            data: {
                archivedBy: parameters?.archivedBy,
                archivedDateRange: parameters?.archivedDateRange,
                issueTypes: parameters?.issueTypes,
                projects: parameters?.projects,
                reporters: parameters?.reporters,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Issues = Issues;
