// jira.js/dist/cjs/version3/jiraExpressions.cjs
'use strict';

class JiraExpressions {
    client;
    constructor(client) {
        this.client = client;
    }
    async analyseExpression(parameters, callback) {
        const config = {
            url: '/rest/api/3/expression/analyse',
            method: 'POST',
            params: {
                check: parameters?.check,
            },
            data: {
                contextVariables: parameters?.contextVariables,
                expressions: parameters?.expressions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async evaluateJiraExpression(parameters, callback) {
        const config = {
            url: '/rest/api/3/expression/eval',
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                context: parameters.context,
                expression: parameters.expression,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async evaluateJiraExpressionUsingEnhancedSearch(parameters, callback) {
        const config = {
            url: '/rest/api/3/expression/evaluate',
            method: 'POST',
            params: {
                expand: parameters.expand,
            },
            data: {
                context: parameters.context,
                expression: parameters.expression,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.JiraExpressions = JiraExpressions;
