// jira.js/dist/cjs/version3/jiraSettings.cjs
'use strict';

class JiraSettings {
    client;
    constructor(client) {
        this.client = client;
    }
    async getApplicationProperty(parameters, callback) {
        const config = {
            url: '/rest/api/3/application-properties',
            method: 'GET',
            params: {
                key: parameters?.key,
                permissionLevel: parameters?.permissionLevel,
                keyFilter: parameters?.keyFilter,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAdvancedSettings(callback) {
        const config = {
            url: '/rest/api/3/application-properties/advanced-settings',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async setApplicationProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/application-properties/${parameters.id}`,
            method: 'PUT',
            data: parameters.body,
        };
        return this.client.sendRequest(config, callback);
    }
    async getConfiguration(callback) {
        const config = {
            url: '/rest/api/3/configuration',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.JiraSettings = JiraSettings;
