// jira.js/dist/cjs/version3/permissions.cjs
'use strict';

class Permissions {
    client;
    constructor(client) {
        this.client = client;
    }
    async getMyPermissions(parameters, callback) {
        const config = {
            url: '/rest/api/3/mypermissions',
            method: 'GET',
            params: {
                projectKey: parameters?.projectKey,
                projectId: parameters?.projectId,
                issueKey: parameters?.issueKey,
                issueId: parameters?.issueId,
                permissions: parameters?.permissions,
                projectUuid: parameters?.projectUuid,
                projectConfigurationUuid: parameters?.projectConfigurationUuid,
                commentId: parameters?.commentId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllPermissions(callback) {
        const config = {
            url: '/rest/api/3/permissions',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getBulkPermissions(parameters, callback) {
        const config = {
            url: '/rest/api/3/permissions/check',
            method: 'POST',
            data: {
                accountId: parameters?.accountId,
                globalPermissions: parameters?.globalPermissions,
                projectPermissions: parameters?.projectPermissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getPermittedProjects(parameters, callback) {
        const config = {
            url: '/rest/api/3/permissions/project',
            method: 'POST',
            data: {
                permissions: parameters?.permissions,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Permissions = Permissions;
