// jira.js/dist/cjs/version3/prioritySchemes.cjs
'use strict';

var paramSerializer = require('../paramSerializer.cjs');

class PrioritySchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getPrioritySchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/priorityscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                priorityId: paramSerializer.paramSerializer('priorityId', parameters?.priorityId),
                schemeId: paramSerializer.paramSerializer('schemeId', parameters?.schemeId),
                schemeName: parameters?.schemeName,
                onlyDefault: parameters?.onlyDefault,
                orderBy: parameters?.orderBy,
                expand: parameters?.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createPriorityScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/priorityscheme',
            method: 'POST',
            data: {
                defaultPriorityId: parameters.defaultPriorityId,
                description: parameters.description,
                mappings: parameters.mappings,
                name: parameters.name,
                priorityIds: parameters.priorityIds,
                projectIds: parameters.projectIds,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async suggestedPrioritiesForMappings(parameters, callback) {
        const config = {
            url: '/rest/api/3/priorityscheme/mappings',
            method: 'POST',
            data: {
                maxResults: parameters?.maxResults,
                priorities: parameters?.priorities,
                projects: parameters?.projects,
                schemeId: parameters?.schemeId,
                startAt: parameters?.startAt,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAvailablePrioritiesByPriorityScheme(parameters, callback) {
        const config = {
            url: '/rest/api/3/priorityscheme/priorities/available',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                query: parameters.query,
                schemeId: parameters.schemeId,
                exclude: parameters.exclude,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updatePriorityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/priorityscheme/${parameters.schemeId}`,
            method: 'PUT',
            data: {
                defaultPriorityId: parameters.defaultPriorityId,
                description: parameters.description,
                mappings: parameters.mappings,
                name: parameters.name,
                priorities: parameters.priorities,
                projects: parameters.projects,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deletePriorityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/priorityscheme/${parameters.schemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
    async getPrioritiesByPriorityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/priorityscheme/${parameters.schemeId}/priorities`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectsByPriorityScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/priorityscheme/${parameters.schemeId}/projects`,
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                projectId: paramSerializer.paramSerializer('projectId', parameters.projectId),
                query: parameters.query,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.PrioritySchemes = PrioritySchemes;
