// jira.js/dist/cjs/version3/projectCategories.cjs
'use strict';

class ProjectCategories {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllProjectCategories(callback) {
        const config = {
            url: '/rest/api/3/projectCategory',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createProjectCategory(parameters, callback) {
        const config = {
            url: '/rest/api/3/projectCategory',
            method: 'POST',
            data: {
                description: parameters.description,
                id: parameters.id,
                name: parameters.name,
                self: parameters.self,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectCategoryById(parameters, callback) {
        const config = {
            url: `/rest/api/3/projectCategory/${parameters.id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateProjectCategory(parameters, callback) {
        const config = {
            url: `/rest/api/3/projectCategory/${parameters.id}`,
            method: 'PUT',
            data: {
                name: parameters.name,
                description: parameters.description,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeProjectCategory(parameters, callback) {
        const config = {
            url: `/rest/api/3/projectCategory/${parameters.id}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectCategories = ProjectCategories;
