// jira.js/dist/cjs/version3/projectClassificationLevels.cjs
'use strict';

class ProjectClassificationLevels {
    client;
    constructor(client) {
        this.client = client;
    }
    async getDefaultProjectClassification(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/classification-level/default`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async updateDefaultProjectClassification(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/classification-level/default`,
            method: 'PUT',
            data: {
                id: parameters.id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeDefaultProjectClassification(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/classification-level/default`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectClassificationLevels = ProjectClassificationLevels;
