// jira.js/dist/cjs/version3/projectTypes.cjs
'use strict';

class ProjectTypes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getAllProjectTypes(callback) {
        const config = {
            url: '/rest/api/3/project/type',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllAccessibleProjectTypes(callback) {
        const config = {
            url: '/rest/api/3/project/type/accessible',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectTypeByKey(parameters, callback) {
        const projectTypeKey = typeof parameters === 'string' ? parameters : parameters.projectTypeKey;
        const config = {
            url: `/rest/api/3/project/type/${projectTypeKey}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getAccessibleProjectTypeByKey(parameters, callback) {
        const projectTypeKey = typeof parameters === 'string' ? parameters : parameters.projectTypeKey;
        const config = {
            url: `/rest/api/3/project/type/${projectTypeKey}/accessible`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectTypes = ProjectTypes;
