// jira.js/dist/cjs/version3/projectTemplates.cjs
'use strict';

class ProjectTemplates {
    client;
    constructor(client) {
        this.client = client;
    }
    async createProjectWithCustomTemplate(parameters, callback) {
        const config = {
            url: '/rest/api/3/project-template',
            method: 'POST',
            data: {
                details: parameters.details,
                template: parameters.template,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async editTemplate(parameters, callback) {
        const config = {
            url: '/rest/api/3/project-template/edit-template',
            method: 'PUT',
            data: {
                templateDescription: parameters.templateDescription,
                templateGenerationOptions: parameters.templateGenerationOptions,
                templateKey: parameters.templateKey,
                templateName: parameters.templateName,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async liveTemplate(parameters, callback) {
        const config = {
            url: '/rest/api/3/project-template/live-template',
            method: 'GET',
            params: {
                projectId: parameters.projectId,
                templateKey: parameters.templateKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async removeTemplate(parameters, callback) {
        const config = {
            url: '/rest/api/3/project-template/remove-template',
            method: 'DELETE',
            params: {
                templateKey: parameters.templateKey,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async saveTemplate(parameters, callback) {
        const config = {
            url: '/rest/api/3/project-template/save-template',
            method: 'POST',
            data: {
                templateDescription: parameters.templateDescription,
                templateFromProjectRequest: parameters.templateFromProjectRequest,
                templateName: parameters.templateName,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectTemplates = ProjectTemplates;
