// jira.js/dist/cjs/version3/projectRoles.cjs
'use strict';

class ProjectRoles {
    client;
    constructor(client) {
        this.client = client;
    }
    async getProjectRoles(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/3/project/${projectIdOrKey}/role`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectRole(parameters, callback) {
        const config = {
            url: `/rest/api/3/project/${parameters.projectIdOrKey}/role/${parameters.id}`,
            method: 'GET',
            params: {
                excludeInactiveUsers: parameters.excludeInactiveUsers,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectRoleDetails(parameters, callback) {
        const projectIdOrKey = typeof parameters === 'string' ? parameters : parameters.projectIdOrKey;
        const config = {
            url: `/rest/api/3/project/${projectIdOrKey}/roledetails`,
            method: 'GET',
            params: {
                currentMember: typeof parameters !== 'string' && parameters.currentMember,
                excludeConnectAddons: typeof parameters !== 'string' && parameters.excludeConnectAddons,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getAllProjectRoles(callback) {
        const config = {
            url: '/rest/api/3/role',
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async createProjectRole(parameters, callback) {
        const config = {
            url: '/rest/api/3/role',
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectRoleById(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/role/${id}`,
            method: 'GET',
        };
        return this.client.sendRequest(config, callback);
    }
    async partialUpdateProjectRole(parameters, callback) {
        const config = {
            url: `/rest/api/3/role/${parameters.id}`,
            method: 'POST',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async fullyUpdateProjectRole(parameters, callback) {
        const config = {
            url: `/rest/api/3/role/${parameters.id}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteProjectRole(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: `/rest/api/3/role/${id}`,
            method: 'DELETE',
            params: {
                swap: typeof parameters !== 'string' && parameters.swap,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ProjectRoles = ProjectRoles;
