// jira.js/dist/cjs/version3/screenSchemes.cjs
'use strict';

class ScreenSchemes {
    client;
    constructor(client) {
        this.client = client;
    }
    async getScreenSchemes(parameters, callback) {
        const config = {
            url: '/rest/api/3/screenscheme',
            method: 'GET',
            params: {
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                id: parameters?.id,
                expand: parameters?.expand,
                queryString: parameters?.queryString,
                orderBy: parameters?.orderBy,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createScreenScheme(parameters, callback) {
        const name = typeof parameters === 'string' ? parameters : parameters.name;
        const config = {
            url: '/rest/api/3/screenscheme',
            method: 'POST',
            data: {
                name,
                description: typeof parameters !== 'string' && parameters.description,
                screens: typeof parameters !== 'string' && parameters.screens,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateScreenScheme(parameters, callback) {
        const config = {
            url: `/rest/api/3/screenscheme/${parameters.screenSchemeId}`,
            method: 'PUT',
            data: {
                description: parameters.description,
                name: parameters.name,
                screens: parameters.screens,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteScreenScheme(parameters, callback) {
        const screenSchemeId = typeof parameters === 'string' ? parameters : parameters.screenSchemeId;
        const config = {
            url: `/rest/api/3/screenscheme/${screenSchemeId}`,
            method: 'DELETE',
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.ScreenSchemes = ScreenSchemes;
