// jira.js/dist/cjs/version3/status.cjs
'use strict';

var pageSchema = require('../schemas/pageSchema.cjs');
require('./models/projectField.cjs');
var jiraStatus = require('./models/jiraStatus.cjs');
require('./models/projectId.cjs');
require('./models/projectIssueTypes.cjs');
require('./models/statusScope.cjs');
var paramSerializer = require('../paramSerializer.cjs');

class Status {
    client;
    constructor(client) {
        this.client = client;
    }
    async getStatusesById(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: '/rest/api/3/statuses',
            method: 'GET',
            params: {
                id,
                expand: typeof parameters !== 'string' ? parameters.expand : undefined,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async createStatuses(parameters, callback) {
        const config = {
            url: '/rest/api/3/statuses',
            method: 'POST',
            data: {
                scope: parameters.scope,
                statuses: parameters.statuses,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateStatuses(parameters, callback) {
        const config = {
            url: '/rest/api/3/statuses',
            method: 'PUT',
            data: {
                statuses: parameters.statuses,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteStatusesById(parameters, callback) {
        const id = typeof parameters === 'string' ? parameters : parameters.id;
        const config = {
            url: '/rest/api/3/statuses',
            method: 'DELETE',
            params: {
                id,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getStatusesByName(parameters, callback) {
        const config = {
            url: '/rest/api/3/statuses/byNames',
            method: 'GET',
            params: {
                name: paramSerializer.paramSerializer('name', parameters.name),
                projectId: parameters.projectId,
            },
        };
        const statuses = await this.client.sendRequest(config, callback);
        return jiraStatus.JiraStatusSchema.array().parse(statuses);
    }
    async search(parameters, callback) {
        const config = {
            url: '/rest/api/3/statuses/search',
            method: 'GET',
            params: {
                expand: parameters?.expand,
                projectId: parameters?.projectId,
                startAt: parameters?.startAt,
                maxResults: parameters?.maxResults,
                searchString: parameters?.searchString,
                statusCategory: parameters?.statusCategory,
            },
        };
        const statuses = await this.client.sendRequest(config, callback);
        return pageSchema.PageSchema(jiraStatus.JiraStatusSchema).parse(statuses);
    }
    async getProjectIssueTypeUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/3/statuses/${parameters.statusId}/project/${parameters.projectId}/issueTypeUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getProjectUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/3/statuses/${parameters.statusId}/projectUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getWorkflowUsagesForStatus(parameters, callback) {
        const config = {
            url: `/rest/api/3/statuses/${parameters.statusId}/workflowUsages`,
            method: 'GET',
            params: {
                nextPageToken: parameters.nextPageToken,
                maxResults: parameters.maxResults,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.Status = Status;
