// jira.js/dist/cjs/version3/userProperties.cjs
'use strict';

class UserProperties {
    client;
    constructor(client) {
        this.client = client;
    }
    async getUserPropertyKeys(parameters, callback) {
        const config = {
            url: '/rest/api/3/user/properties',
            method: 'GET',
            params: {
                accountId: parameters?.accountId,
                userKey: parameters?.userKey,
                username: parameters?.username,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async getUserProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/user/properties/${parameters.propertyKey}`,
            method: 'GET',
            params: {
                accountId: parameters.accountId,
                userKey: parameters.userKey,
                username: parameters.username,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async setUserProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/user/properties/${parameters.propertyKey}`,
            method: 'PUT',
            params: {
                accountId: parameters.accountId,
            },
            data: parameters.propertyValue,
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteUserProperty(parameters, callback) {
        const config = {
            url: `/rest/api/3/user/properties/${parameters.propertyKey}`,
            method: 'DELETE',
            params: {
                accountId: parameters.accountId,
                userKey: parameters.userKey,
                username: parameters.username,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.UserProperties = UserProperties;
