// jira.js/dist/cjs/version3/workflowSchemeProjectAssociations.cjs
'use strict';

var paramSerializer = require('../paramSerializer.cjs');

class WorkflowSchemeProjectAssociations {
    client;
    constructor(client) {
        this.client = client;
    }
    async getWorkflowSchemeProjectAssociations(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/project',
            method: 'GET',
            params: {
                projectId: paramSerializer.paramSerializer('projectId', parameters.projectId),
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async assignSchemeToProject(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflowscheme/project',
            method: 'PUT',
            data: {
                projectId: parameters.projectId,
                workflowSchemeId: parameters.workflowSchemeId,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.WorkflowSchemeProjectAssociations = WorkflowSchemeProjectAssociations;
