// jira.js/dist/cjs/version3/workflowTransitionRules.cjs
'use strict';

class WorkflowTransitionRules {
    client;
    constructor(client) {
        this.client = client;
    }
    async getWorkflowTransitionRuleConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflow/rule/config',
            method: 'GET',
            params: {
                startAt: parameters.startAt,
                maxResults: parameters.maxResults,
                types: parameters.types,
                keys: parameters.keys,
                workflowNames: parameters.workflowNames,
                withTags: parameters.withTags,
                draft: parameters.draft,
                expand: parameters.expand,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async updateWorkflowTransitionRuleConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflow/rule/config',
            method: 'PUT',
            data: {
                workflows: parameters.workflows,
            },
        };
        return this.client.sendRequest(config, callback);
    }
    async deleteWorkflowTransitionRuleConfigurations(parameters, callback) {
        const config = {
            url: '/rest/api/3/workflow/rule/config/delete',
            method: 'PUT',
            data: {
                workflows: parameters?.workflows,
            },
        };
        return this.client.sendRequest(config, callback);
    }
}

exports.WorkflowTransitionRules = WorkflowTransitionRules;
